diff --git a/src/test/java/com/headius/invokebinder/SignatureTest.java b/src/test/java/com/headius/invokebinder/SignatureTest.java
index a5a270f..e1a7806 100644
--- a/src/test/java/com/headius/invokebinder/SignatureTest.java
+++ b/src/test/java/com/headius/invokebinder/SignatureTest.java
@@ -376,6 +376,7 @@ public class SignatureTest {
         assertEquals(Integer[].class, newSig.argType(1));
         assertEquals("bs", newSig.argName(1));
         assertEquals(3, newSig.argCount());
+        assertEquals("c", newSig.argName(2));
 
         oldSig = Subjects.StringIntegerIntegerInteger;
         newSig = oldSig.collect("bs", "b.*");
