diff --git a/src/test/java/com/vdurmont/semver4j/NpmSemverTest.java b/src/test/java/com/vdurmont/semver4j/NpmSemverTest.java
index 3d0a097..30d7789 100644
--- a/src/test/java/com/vdurmont/semver4j/NpmSemverTest.java
+++ b/src/test/java/com/vdurmont/semver4j/NpmSemverTest.java
@@ -176,12 +176,12 @@ public class NpmSemverTest {
             // Complex ranges:
             // TODO: Add more!
 
-            // (66) TODO
-            // { "1.2.0", "1.2 <1.2.8 || >2.0.0", true, },
-            // (67) TODO
-            // { "1.2.7", "1.2 <1.2.8 || >2.0.0", true, },
-            // (68) TODO
-            // { "2.0.1", "1.2 <1.2.8 || >2.0.0", true, },
+            // (66)
+             { "1.2.0", "1.2 <1.2.8 || >2.0.0", true, },
+            // (67)
+             { "1.2.7", "1.2 <1.2.8 || >2.0.0", true, },
+            // (68)
+             { "2.0.1", "1.2 <1.2.8 || >2.0.0", true, },
             // (69)
             { "1.1.0", "1.2 <1.2.8 || >2.0.0", false, },
             // (70)
