diff --git a/src/test/java/gridss/kraken/ExtractBestSequencesBasedOnReportTest.java b/src/test/java/gridss/kraken/ExtractBestSequencesBasedOnReportTest.java
index c3a9efd..206414f 100644
--- a/src/test/java/gridss/kraken/ExtractBestSequencesBasedOnReportTest.java
+++ b/src/test/java/gridss/kraken/ExtractBestSequencesBasedOnReportTest.java
@@ -181,7 +181,7 @@ public class ExtractBestSequencesBasedOnReportTest extends IntermediateFilesTest
         cmd.doWork();
 
         List<String> summary = Files.readAllLines(cmd.SUMMARY_OUTPUT.toPath());
-        Assert.assertEquals("10405\tOrthohepadnavirus\t858\t10407\tHepatitis B virus\t856\t489466\tHBV genotype C\t277\tkraken:taxid|489466|AB540583", summary.get(1));
+        Assert.assertEquals("10405\tOrthohepadnavirus\t858\t10407\tHepatitis B virus\t856\t489466\tHBV genotype C\t277\t276\tkraken:taxid|489466|AB540583", summary.get(1));
     }
     //@Test
     @Ignore // Underlying issue was that HPV-45 isn't in ReqSeq and the first half of VIRUSBreakend was run against a larger database
diff --git a/src/test/resources/kraken/multistrain/26T.virusbreakend.vcf.kraken2.report.all.txt b/src/test/resources/kraken/multistrain/26T.virusbreakend.vcf.kraken2.report.all.txt
index 9c3f06f..f0cfe5a 100644
--- a/src/test/resources/kraken/multistrain/26T.virusbreakend.vcf.kraken2.report.all.txt
+++ b/src/test/resources/kraken/multistrain/26T.virusbreakend.vcf.kraken2.report.all.txt
@@ -40,7 +40,7 @@
   0.00	858	0	F	10404	              Hepadnaviridae
   0.00	858	2	G	10405	                Orthohepadnavirus
   0.00	856	271	S	10407	                  Hepatitis B virus
-  0.00	277	277	S1	489466	                    HBV genotype C
+  0.00	277	276	S1	489466	                    HBV genotype C
   0.00	86	0	S1	621278	                    recombinant Hepatitis B virus
   0.00	74	74	S2	621277	                      Hepatitis B virus 146/Thailand
   0.00	12	12	S2	1282342	                      Hepatitis B virus subgenotype A3/genotype E
