diff --git a/src/main/java/org/traccar/protocol/Minifinder2ProtocolDecoder.java b/src/main/java/org/traccar/protocol/Minifinder2ProtocolDecoder.java
index 0b08bad..aa43a60 100644
--- a/src/main/java/org/traccar/protocol/Minifinder2ProtocolDecoder.java
+++ b/src/main/java/org/traccar/protocol/Minifinder2ProtocolDecoder.java
@@ -277,6 +277,10 @@ public class Minifinder2ProtocolDecoder extends BaseProtocolDecoder {
                             i += 1;
                         }
                         break;
+                    case 0x37:
+                        buf.readUnsignedIntLE(); // timestamp
+                        position.set("barkCount", BitUtil.to(buf.readUnsignedIntLE(), 31));
+                        break;
                     case 0x40:
                         buf.readUnsignedIntLE(); // timestamp
                         int heartRate = buf.readUnsignedByte();
