diff --git a/src/main/java/org/osm2world/core/world/modules/building/LevelAndHeightData.java b/src/main/java/org/osm2world/core/world/modules/building/LevelAndHeightData.java
index 7be7d16..7562154 100644
--- a/src/main/java/org/osm2world/core/world/modules/building/LevelAndHeightData.java
+++ b/src/main/java/org/osm2world/core/world/modules/building/LevelAndHeightData.java
@@ -241,10 +241,12 @@ public class LevelAndHeightData {
 		Function<Integer, LevelType> getLevelType = level -> {
 			int index = levelNumbers.indexOf(level);
 			if (index < buildingUndergroundLevels) return LevelType.UNDERGROUND;
-			if (index < buildingUndergroundLevels + buildingLevels) return LevelType.ABOVEGROUND;
+			if (index < buildingUndergroundLevels + (buildingLevels - buildingMinLevel)) return LevelType.ABOVEGROUND;
 			return LevelType.ROOF;
 		};
 
+
+
 		List<Integer> levelNumbersAboveground = levelNumbers.stream().filter(l -> getLevelType.apply(l) == LevelType.ABOVEGROUND).collect(toList());
 		List<Integer> levelNumbersRoof = levelNumbers.stream().filter(l -> getLevelType.apply(l) == LevelType.ROOF).collect(toList());
 
