diff --git a/expr_test.go b/expr_test.go
index 6fc4d5a..0cc0aa1 100644
--- a/expr_test.go
+++ b/expr_test.go
@@ -323,7 +323,7 @@ var testTable = []string{
 	(`1 != 2 > 1 != 1`), ("true"),
 	(`1 != 2 < 1 != 1`), ("false"),
 	(`1 != 1 < 2 != 1`), ("true"),
-	(`u64+"hello"`), ("[Function u64]hello"),
+	(`u64+"hello"`), ("[Function: u64]hello"),
 	(`0.123123i64`), (`SyntaxError`),
 	(`new`), (`SyntaxError`),
 	(`howdy.myfn1().myfn2("1",2,"3") == 6`), (`true`),
@@ -342,8 +342,8 @@ var testTable = []string{
 	(`((0u64)%0u64)`), (`NaN`),
 	(`((0u64)/0u64)`), (`NaN`),
 	(`64`), (`64`),
-	(`u64`), (`[Function u64]`),
-	(`i64`), (`[Function i64]`),
+	(`u64`), (`[Function: u64]`),
+	(`i64`), (`[Function: i64]`),
 	(`1 == "1"`), (`true`),
 	(`1 === "1"`), (`false`),
 	(`1 !== "1"`), (`true`),
