diff --git a/src/main/java/org/jfree/data/Range.java b/src/main/java/org/jfree/data/Range.java
index fa28cb5..ff3f860 100644
--- a/src/main/java/org/jfree/data/Range.java
+++ b/src/main/java/org/jfree/data/Range.java
@@ -191,7 +191,10 @@ public strictfp class Range implements Serializable {
         if (value > this.upper) {
             return this.upper;
         }
-        return this.lower;
+        if (value < this.lower) {
+            return this.lower;
+        }
+        return value; // covers Double.NaN
     }
 
     /**
