diff --git a/src/test/java/com/example/authorizationserver/oauth/endpoint/AuthorizationEndpointIntegrationTest.java b/src/test/java/com/example/authorizationserver/oauth/endpoint/AuthorizationEndpointIntegrationTest.java
index 5beee9f..bdec78d 100644
--- a/src/test/java/com/example/authorizationserver/oauth/endpoint/AuthorizationEndpointIntegrationTest.java
+++ b/src/test/java/com/example/authorizationserver/oauth/endpoint/AuthorizationEndpointIntegrationTest.java
@@ -49,7 +49,7 @@ class AuthorizationEndpointIntegrationTest {
     var multiValueMap = new LinkedMultiValueMap<String, String>();
     multiValueMap.add("response_type", "code");
     multiValueMap.add("scope", "openid");
-    multiValueMap.add("client_id", "confidential-demo");
+    multiValueMap.add("client_id", "confidential-jwt");
     multiValueMap.add("redirect_uri", "http://localhost:9090/demo-client/login/oauth2/code/demo");
     this.mockMvc
         .perform(
@@ -67,7 +67,7 @@ class AuthorizationEndpointIntegrationTest {
     var multiValueMap = new LinkedMultiValueMap<String, String>();
     multiValueMap.add("response_type", "code");
     multiValueMap.add("scope", "openid");
-    multiValueMap.add("client_id", "public-demo");
+    multiValueMap.add("client_id", "public-jwt");
     multiValueMap.add("code_challenge", "123456789");
     multiValueMap.add("redirect_uri", "http://localhost:9090/demo-client/login/oauth2/code/demo");
     this.mockMvc
@@ -86,7 +86,7 @@ class AuthorizationEndpointIntegrationTest {
     var multiValueMap = new LinkedMultiValueMap<String, String>();
     multiValueMap.add("response_type", "code");
     multiValueMap.add("scope", "openid");
-    multiValueMap.add("client_id", "public-demo");
+    multiValueMap.add("client_id", "public-jwt");
     multiValueMap.add("redirect_uri", "http://localhost:9090/demo-client/login/oauth2/code/demo");
     this.mockMvc
             .perform(
@@ -176,7 +176,7 @@ class AuthorizationEndpointIntegrationTest {
     var multiValueMap = new LinkedMultiValueMap<String, String>();
     multiValueMap.add("response_type", "code");
     multiValueMap.add("scope", "openid");
-    multiValueMap.add("client_id", "confidential-demo");
+    multiValueMap.add("client_id", "confidential-jwt");
     multiValueMap.add("redirect_uri", "http://localhost:9090/dummy");
     this.mockMvc
             .perform(
