diff --git a/src/test/java/com/github/pedrovgs/problem12/MoveZerosInArrayTest.java b/src/test/java/com/github/pedrovgs/problem12/MoveZerosInArrayTest.java
index 2865ed0..99f8562 100644
--- a/src/test/java/com/github/pedrovgs/problem12/MoveZerosInArrayTest.java
+++ b/src/test/java/com/github/pedrovgs/problem12/MoveZerosInArrayTest.java
@@ -44,6 +44,14 @@ public class MoveZerosInArrayTest {
     assertArrayEquals(new int[0], array);
   }
 
+  @Test public void shouldWorkWithPositiveNegativeZerosArraySorting() {
+    int[] array = {-1, 0, 2, 4, 0, -3, -5, 0, 6, -3, 0};
+
+    moveZeros.moveSorting(array);
+
+    assertZerosAtRight(array);
+  }
+
   @Test public void shouldOrganizeAnArrayFullOfZerosSorting() {
     int[] array = { 0, 0, 0, 0, 0 };
 
