diff --git a/src/main/java/org/codehaus/gmavenplus/model/Version.java b/src/main/java/org/codehaus/gmavenplus/model/Version.java
index 9c1acd3..b4c9271 100644
--- a/src/main/java/org/codehaus/gmavenplus/model/Version.java
+++ b/src/main/java/org/codehaus/gmavenplus/model/Version.java
@@ -238,7 +238,8 @@ public class Version implements Comparable<Version> {
         }
         if (comp == 0) {
             if (tag != null && version.tag != null) {
-                return tag.replace("beta", " beta").compareTo(version.tag.replace("beta", " beta"));
+                return tag.replace("beta", " beta").replace("alpha", " alpha")
+                        .compareTo(version.tag.replace("beta", " beta").replace("alpha", " alpha"));
             } else if (tag == null ^ version.tag == null) {
                 if (tag == null) {
                     return noTagsAreNewer ? 1 : -1;
