diff --git a/dotenv/godotenv_test.go b/dotenv/godotenv_test.go
index da2265c..163941d 100644
--- a/dotenv/godotenv_test.go
+++ b/dotenv/godotenv_test.go
@@ -718,3 +718,11 @@ func TestUTF8BOM(t *testing.T) {
 
 	loadEnvAndCompareValues(t, Load, envFileName, expectedValues, noopPresets)
 }
+
+func TestDash(t *testing.T) {
+	loadEnvAndCompareValues(t, Load, "fixtures/special.env", map[string]string{
+		"VAR-WITH-DASHES":      "dashes",
+		"VAR.WITH.DOTS":        "dots",
+		"VAR_WITH_UNDERSCORES": "underscores",
+	}, noopPresets)
+}
