diff --git a/src/test/java/com/vonage/client/meetings/CreateRoomEndpointTest.java b/src/test/java/com/vonage/client/meetings/CreateRoomEndpointTest.java
index ae6698c..2215916 100644
--- a/src/test/java/com/vonage/client/meetings/CreateRoomEndpointTest.java
+++ b/src/test/java/com/vonage/client/meetings/CreateRoomEndpointTest.java
@@ -67,7 +67,7 @@ public class CreateRoomEndpointTest {
 
 		RequestBuilder builder = endpoint.makeRequest(request);
 		assertEquals("POST", builder.getMethod());
-		String expectedUri = "https://api-eu.vonage.com/meetings/rooms";
+		String expectedUri = "https://api-eu.vonage.com/v1/meetings/rooms";
 		assertEquals(expectedUri, builder.build().getURI().toString());
 		assertEquals(ContentType.APPLICATION_JSON.getMimeType(), builder.getFirstHeader("Content-Type").getValue());
 
@@ -107,7 +107,7 @@ public class CreateRoomEndpointTest {
 		String baseUri = "http://example.com";
 		HttpWrapper wrapper = new HttpWrapper(HttpConfig.builder().baseUri(baseUri).build());
 		endpoint = new CreateRoomEndpoint(wrapper);
-		String expectedUri = baseUri + "/meetings/rooms";
+		String expectedUri = baseUri + "/v1/meetings/rooms";
 		String displayName = "My custom room";
 		MeetingRoom request = MeetingRoom.builder(displayName).build();
 		RequestBuilder builder = endpoint.makeRequest(request);
diff --git a/src/test/java/com/vonage/client/meetings/CreateThemeEndpointTest.java b/src/test/java/com/vonage/client/meetings/CreateThemeEndpointTest.java
index dff9354..3e742d2 100644
--- a/src/test/java/com/vonage/client/meetings/CreateThemeEndpointTest.java
+++ b/src/test/java/com/vonage/client/meetings/CreateThemeEndpointTest.java
@@ -42,7 +42,7 @@ public class CreateThemeEndpointTest {
 
 		RequestBuilder builder = endpoint.makeRequest(request);
 		assertEquals("POST", builder.getMethod());
-		String expectedUri = "https://api-eu.vonage.com/meetings/themes";
+		String expectedUri = "https://api-eu.vonage.com/v1/meetings/themes";
 		assertEquals(expectedUri, builder.build().getURI().toString());
 		assertEquals(ContentType.APPLICATION_JSON.getMimeType(), builder.getFirstHeader("Content-Type").getValue());
 		assertEquals(ContentType.APPLICATION_JSON.getMimeType(), builder.getFirstHeader("Accept").getValue());
@@ -66,7 +66,7 @@ public class CreateThemeEndpointTest {
 		String baseUri = "http://example.com";
 		HttpWrapper wrapper = new HttpWrapper(HttpConfig.builder().baseUri(baseUri).build());
 		endpoint = new CreateThemeEndpoint(wrapper);
-		String expectedUri = baseUri + "/meetings/themes";
+		String expectedUri = baseUri + "/v1/meetings/themes";
 		Theme request = Theme.builder().build();
 		RequestBuilder builder = endpoint.makeRequest(request);
 		assertEquals(expectedUri, builder.build().getURI().toString());
diff --git a/src/test/java/com/vonage/client/meetings/DeleteRecordingEndpointTest.java b/src/test/java/com/vonage/client/meetings/DeleteRecordingEndpointTest.java
index 067b8ac..9fb11a1 100644
--- a/src/test/java/com/vonage/client/meetings/DeleteRecordingEndpointTest.java
+++ b/src/test/java/com/vonage/client/meetings/DeleteRecordingEndpointTest.java
@@ -39,7 +39,7 @@ public class DeleteRecordingEndpointTest {
 		UUID recordingId = UUID.randomUUID();
 		RequestBuilder builder = endpoint.makeRequest(recordingId);
 		assertEquals("DELETE", builder.getMethod());
-		String expectedUri = "https://api-eu.vonage.com/meetings/recordings/"+recordingId;
+		String expectedUri = "https://api-eu.vonage.com/v1/meetings/recordings/"+recordingId;
 		assertEquals(expectedUri, builder.build().getURI().toString());
 		assertNull(endpoint.parseResponse(TestUtils.makeJsonHttpResponse(204, "")));
 	}
@@ -50,7 +50,7 @@ public class DeleteRecordingEndpointTest {
 		UUID recordingId = UUID.randomUUID();
 		HttpWrapper wrapper = new HttpWrapper(HttpConfig.builder().baseUri(baseUri).build());
 		endpoint = new DeleteRecordingEndpoint(wrapper);
-		String expectedUri = baseUri + "/meetings/recordings/"+recordingId;
+		String expectedUri = baseUri + "/v1/meetings/recordings/"+recordingId;
 		RequestBuilder builder = endpoint.makeRequest(recordingId);
 		assertEquals(expectedUri, builder.build().getURI().toString());
 		assertEquals("DELETE", builder.getMethod());
diff --git a/src/test/java/com/vonage/client/meetings/DeleteThemeEndpointTest.java b/src/test/java/com/vonage/client/meetings/DeleteThemeEndpointTest.java
index 1eaf34c..2e6aee8 100644
--- a/src/test/java/com/vonage/client/meetings/DeleteThemeEndpointTest.java
+++ b/src/test/java/com/vonage/client/meetings/DeleteThemeEndpointTest.java
@@ -39,7 +39,7 @@ public class DeleteThemeEndpointTest {
 		UUID themeId = UUID.randomUUID();
 		RequestBuilder builder = endpoint.makeRequest(new DeleteThemeRequest(themeId, false));
 		assertEquals("DELETE", builder.getMethod());
-		String expectedUri = "https://api-eu.vonage.com/meetings/themes/"+themeId+"?force=false";
+		String expectedUri = "https://api-eu.vonage.com/v1/meetings/themes/"+themeId+"?force=false";
 		assertEquals(expectedUri, builder.build().getURI().toString());
 		assertNull(endpoint.parseResponse(TestUtils.makeJsonHttpResponse(204, "")));
 	}
@@ -50,7 +50,7 @@ public class DeleteThemeEndpointTest {
 		String baseUri = "http://example.com";
 		HttpWrapper wrapper = new HttpWrapper(HttpConfig.builder().baseUri(baseUri).build());
 		endpoint = new DeleteThemeEndpoint(wrapper);
-		String expectedUri = baseUri + "/meetings/themes/"+themeId+"?force=true";
+		String expectedUri = baseUri + "/v1/meetings/themes/"+themeId+"?force=true";
 		RequestBuilder builder = endpoint.makeRequest(new DeleteThemeRequest(themeId, true));
 		assertEquals(expectedUri, builder.build().getURI().toString());
 		assertEquals("DELETE", builder.getMethod());
diff --git a/src/test/java/com/vonage/client/meetings/FinalizeLogosEndpointTest.java b/src/test/java/com/vonage/client/meetings/FinalizeLogosEndpointTest.java
index 411d1ec..490db4a 100644
--- a/src/test/java/com/vonage/client/meetings/FinalizeLogosEndpointTest.java
+++ b/src/test/java/com/vonage/client/meetings/FinalizeLogosEndpointTest.java
@@ -43,7 +43,7 @@ public class FinalizeLogosEndpointTest {
 		FinalizeLogosRequest request = new FinalizeLogosRequest(themeId, Arrays.asList("col", "fff"));
 		RequestBuilder builder = endpoint.makeRequest(request);
 		assertEquals("PUT", builder.getMethod());
-		String expectedUri = "https://api-eu.vonage.com/meetings/themes/"+themeId+"/finalizeLogos";
+		String expectedUri = "https://api-eu.vonage.com/v1/meetings/themes/"+themeId+"/finalizeLogos";
 		assertEquals(expectedUri, builder.build().getURI().toString());
 		assertNull(endpoint.parseResponse(TestUtils.makeJsonHttpResponse(200, "")));
 	}
@@ -54,7 +54,7 @@ public class FinalizeLogosEndpointTest {
 		String baseUri = "http://example.com";
 		HttpWrapper wrapper = new HttpWrapper(HttpConfig.builder().baseUri(baseUri).build());
 		endpoint = new FinalizeLogosEndpoint(wrapper);
-		String expectedUri = baseUri + "/meetings/themes/"+themeId+"/finalizeLogos";
+		String expectedUri = baseUri + "/v1/meetings/themes/"+themeId+"/finalizeLogos";
 		FinalizeLogosRequest request = new FinalizeLogosRequest(themeId, Arrays.asList("lk1", "lk2"));
 		RequestBuilder builder = endpoint.makeRequest(request);
 		assertEquals(expectedUri, builder.build().getURI().toString());
diff --git a/src/test/java/com/vonage/client/meetings/GetLogoUploadUrlsEndpointTest.java b/src/test/java/com/vonage/client/meetings/GetLogoUploadUrlsEndpointTest.java
index a6b85f4..0e12d62 100644
--- a/src/test/java/com/vonage/client/meetings/GetLogoUploadUrlsEndpointTest.java
+++ b/src/test/java/com/vonage/client/meetings/GetLogoUploadUrlsEndpointTest.java
@@ -39,7 +39,7 @@ public class GetLogoUploadUrlsEndpointTest {
 	public void testDefaultUri() throws Exception {
 		RequestBuilder builder = endpoint.makeRequest(null);
 		assertEquals("GET", builder.getMethod());
-		String expectedUri = "https://api-eu.vonage.com/meetings/themes/logos-upload-urls";
+		String expectedUri = "https://api-eu.vonage.com/v1/meetings/themes/logos-upload-urls";
 		assertEquals(expectedUri, builder.build().getURI().toString());
 	}
 
@@ -48,7 +48,7 @@ public class GetLogoUploadUrlsEndpointTest {
 		String baseUri = "http://example.com";
 		HttpWrapper wrapper = new HttpWrapper(HttpConfig.builder().baseUri(baseUri).build());
 		endpoint = new GetLogoUploadUrlsEndpoint(wrapper);
-		String expectedUri = baseUri + "/meetings/themes/logos-upload-urls";
+		String expectedUri = baseUri + "/v1/meetings/themes/logos-upload-urls";
 		RequestBuilder builder = endpoint.makeRequest(null);
 		assertEquals(expectedUri, builder.build().getURI().toString());
 		assertEquals("GET", builder.getMethod());
diff --git a/src/test/java/com/vonage/client/meetings/GetRecordingEndpointTest.java b/src/test/java/com/vonage/client/meetings/GetRecordingEndpointTest.java
index 0fe9ba5..40a7a37 100644
--- a/src/test/java/com/vonage/client/meetings/GetRecordingEndpointTest.java
+++ b/src/test/java/com/vonage/client/meetings/GetRecordingEndpointTest.java
@@ -41,7 +41,7 @@ public class GetRecordingEndpointTest {
 		UUID recordingId = UUID.randomUUID();
 		RequestBuilder builder = endpoint.makeRequest(recordingId);
 		assertEquals("GET", builder.getMethod());
-		String expectedUri = "https://api-eu.vonage.com/meetings/recordings/"+recordingId;
+		String expectedUri = "https://api-eu.vonage.com/v1/meetings/recordings/"+recordingId;
 		assertEquals(expectedUri, builder.build().getURI().toString());
 		assertEquals(ContentType.APPLICATION_JSON.getMimeType(), builder.getFirstHeader("Accept").getValue());
 		String expectedResponse = "{\"nonsense\":0,\"_links\":{\"url\":{}}}";
@@ -62,7 +62,7 @@ public class GetRecordingEndpointTest {
 		String baseUri = "http://example.com";
 		HttpWrapper wrapper = new HttpWrapper(HttpConfig.builder().baseUri(baseUri).build());
 		endpoint = new GetRecordingEndpoint(wrapper);
-		String expectedUri = baseUri + "/meetings/recordings/"+recordingId;
+		String expectedUri = baseUri + "/v1/meetings/recordings/"+recordingId;
 		RequestBuilder builder = endpoint.makeRequest(recordingId);
 		assertEquals(expectedUri, builder.build().getURI().toString());
 		assertEquals(ContentType.APPLICATION_JSON.getMimeType(), builder.getFirstHeader("Accept").getValue());
diff --git a/src/test/java/com/vonage/client/meetings/GetRoomEndpointTest.java b/src/test/java/com/vonage/client/meetings/GetRoomEndpointTest.java
index 8e6dbd2..eb590f3 100644
--- a/src/test/java/com/vonage/client/meetings/GetRoomEndpointTest.java
+++ b/src/test/java/com/vonage/client/meetings/GetRoomEndpointTest.java
@@ -40,7 +40,7 @@ public class GetRoomEndpointTest {
 		UUID roomId = UUID.randomUUID();
 		RequestBuilder builder = endpoint.makeRequest(roomId);
 		assertEquals("GET", builder.getMethod());
-		String expectedUri = "https://api-eu.vonage.com/meetings/rooms/"+roomId;
+		String expectedUri = "https://api-eu.vonage.com/v1/meetings/rooms/"+roomId;
 		assertEquals(expectedUri, builder.build().getURI().toString());
 		assertEquals(ContentType.APPLICATION_JSON.getMimeType(), builder.getFirstHeader("Accept").getValue());
 		HttpResponse mockResponse = TestUtils.makeJsonHttpResponse(200, MeetingsClientTest.SAMPLE_ROOM_RESPONSE);
@@ -54,7 +54,7 @@ public class GetRoomEndpointTest {
 		String baseUri = "https://example.com";
 		HttpWrapper wrapper = new HttpWrapper(HttpConfig.builder().baseUri(baseUri).build());
 		endpoint = new GetRoomEndpoint(wrapper);
-		String expectedUri = baseUri + "/meetings/rooms/"+roomId;
+		String expectedUri = baseUri + "/v1/meetings/rooms/"+roomId;
 		RequestBuilder builder = endpoint.makeRequest(roomId);
 		assertEquals(expectedUri, builder.build().getURI().toString());
 		assertEquals(ContentType.APPLICATION_JSON.getMimeType(), builder.getFirstHeader("Accept").getValue());
diff --git a/src/test/java/com/vonage/client/meetings/GetThemeEndpointTest.java b/src/test/java/com/vonage/client/meetings/GetThemeEndpointTest.java
index 1cfde65..67839f8 100644
--- a/src/test/java/com/vonage/client/meetings/GetThemeEndpointTest.java
+++ b/src/test/java/com/vonage/client/meetings/GetThemeEndpointTest.java
@@ -41,7 +41,7 @@ public class GetThemeEndpointTest {
 		UUID themeId = UUID.randomUUID();
 		RequestBuilder builder = endpoint.makeRequest(themeId);
 		assertEquals("GET", builder.getMethod());
-		String expectedUri = "https://api-eu.vonage.com/meetings/themes/"+themeId;
+		String expectedUri = "https://api-eu.vonage.com/v1/meetings/themes/"+themeId;
 		assertEquals(expectedUri, builder.build().getURI().toString());
 		assertEquals(ContentType.APPLICATION_JSON.getMimeType(), builder.getFirstHeader("Accept").getValue());
 		String expectedResponse = "{\"nonsense\":true}";
@@ -56,7 +56,7 @@ public class GetThemeEndpointTest {
 		HttpWrapper wrapper = new HttpWrapper(HttpConfig.builder().baseUri(baseUri).build());
 		endpoint = new GetThemeEndpoint(wrapper);
 		UUID themeId = UUID.randomUUID();
-		String expectedUri = baseUri + "/meetings/themes/"+themeId;
+		String expectedUri = baseUri + "/v1/meetings/themes/"+themeId;
 		RequestBuilder builder = endpoint.makeRequest(themeId);
 		assertEquals(expectedUri, builder.build().getURI().toString());
 		assertEquals(ContentType.APPLICATION_JSON.getMimeType(), builder.getFirstHeader("Accept").getValue());
diff --git a/src/test/java/com/vonage/client/meetings/ListDialNumbersEndpointTest.java b/src/test/java/com/vonage/client/meetings/ListDialNumbersEndpointTest.java
index bc8605c..af3a825 100644
--- a/src/test/java/com/vonage/client/meetings/ListDialNumbersEndpointTest.java
+++ b/src/test/java/com/vonage/client/meetings/ListDialNumbersEndpointTest.java
@@ -39,7 +39,7 @@ public class ListDialNumbersEndpointTest {
 	public void testDefaultUri() throws Exception {
 		RequestBuilder builder = endpoint.makeRequest(null);
 		assertEquals("GET", builder.getMethod());
-		String expectedUri = "https://api-eu.vonage.com/meetings/dial-in-numbers";
+		String expectedUri = "https://api-eu.vonage.com/v1/meetings/dial-in-numbers";
 		assertEquals(expectedUri, builder.build().getURI().toString());
 		String expectedPayload = "[\n" +
 				"{},   {\n" +
@@ -75,7 +75,7 @@ public class ListDialNumbersEndpointTest {
 		String baseUri = "http://example.com";
 		HttpWrapper wrapper = new HttpWrapper(HttpConfig.builder().baseUri(baseUri).build());
 		endpoint = new ListDialNumbersEndpoint(wrapper);
-		String expectedUri = baseUri + "/meetings/dial-in-numbers";
+		String expectedUri = baseUri + "/v1/meetings/dial-in-numbers";
 		RequestBuilder builder = endpoint.makeRequest(null);
 		assertEquals(expectedUri, builder.build().getURI().toString());
 		assertEquals("GET", builder.getMethod());
diff --git a/src/test/java/com/vonage/client/meetings/ListRecordingsEndpointTest.java b/src/test/java/com/vonage/client/meetings/ListRecordingsEndpointTest.java
index ebbe0a3..3ab67ad 100644
--- a/src/test/java/com/vonage/client/meetings/ListRecordingsEndpointTest.java
+++ b/src/test/java/com/vonage/client/meetings/ListRecordingsEndpointTest.java
@@ -40,7 +40,7 @@ public class ListRecordingsEndpointTest {
 		String sessionId = "2_MX40NjMwODczMn5-MTU3NTgyODEwNzQ2M...";
 		RequestBuilder builder = endpoint.makeRequest(sessionId);
 		assertEquals("GET", builder.getMethod());
-		String expectedUri = "https://api-eu.vonage.com/meetings/sessions/"+sessionId+"/recordings";
+		String expectedUri = "https://api-eu.vonage.com/v1/meetings/sessions/"+sessionId+"/recordings";
 		assertEquals(expectedUri, builder.build().getURI().toString());
 		assertEquals(ContentType.APPLICATION_JSON.getMimeType(), builder.getFirstHeader("Accept").getValue());
 		String expectedPayload = "{\"_embedded\":{\"recordings\":[{},{\"_links\":{},\"status\":\"uploaded\"}]}}";
@@ -65,7 +65,7 @@ public class ListRecordingsEndpointTest {
 		String sessionId = "2_MX40NjMwODczMn5-MTU3NTgyODEwNzQ2Jk.." , baseUri = "http://example.com";
 		HttpWrapper wrapper = new HttpWrapper(HttpConfig.builder().baseUri(baseUri).build());
 		endpoint = new ListRecordingsEndpoint(wrapper);
-		String expectedUri = baseUri + "/meetings/sessions/"+sessionId+"/recordings";
+		String expectedUri = baseUri + "/v1/meetings/sessions/"+sessionId+"/recordings";
 		RequestBuilder builder = endpoint.makeRequest(sessionId);
 		assertEquals(expectedUri, builder.build().getURI().toString());
 		assertEquals(ContentType.APPLICATION_JSON.getMimeType(), builder.getFirstHeader("Accept").getValue());
diff --git a/src/test/java/com/vonage/client/meetings/ListRoomsEndpointTest.java b/src/test/java/com/vonage/client/meetings/ListRoomsEndpointTest.java
index 63c58cd..cdf52ca 100644
--- a/src/test/java/com/vonage/client/meetings/ListRoomsEndpointTest.java
+++ b/src/test/java/com/vonage/client/meetings/ListRoomsEndpointTest.java
@@ -41,7 +41,7 @@ public class ListRoomsEndpointTest {
 		ListRoomsRequest request = new ListRoomsRequest(null, null, null, null);
 		RequestBuilder builder = endpoint.makeRequest(request);
 		assertEquals("GET", builder.getMethod());
-		String expectedUri = "https://api-eu.vonage.com/meetings/rooms";
+		String expectedUri = "https://api-eu.vonage.com/v1/meetings/rooms";
 		assertEquals(expectedUri, builder.build().getURI().toString());
 		Map<String, String> params = TestUtils.makeParameterMap(builder.getParameters());
 		assertEquals(0, params.size());
@@ -62,7 +62,7 @@ public class ListRoomsEndpointTest {
 		HttpWrapper wrapper = new HttpWrapper(HttpConfig.builder().baseUri(baseUri).build());
 		endpoint = new ListRoomsEndpoint(wrapper);
 		ListRoomsRequest request = new ListRoomsRequest(51, 150, 25, null);
-		String expectedUri = baseUri + "/meetings/rooms?" +
+		String expectedUri = baseUri + "/v1/meetings/rooms?" +
 				"start_id="+request.startId+"&end_id="+request.endId+"&page_size="+request.pageSize;
 		RequestBuilder builder = endpoint.makeRequest(request);
 		assertEquals(expectedUri, builder.build().getURI().toString());
diff --git a/src/test/java/com/vonage/client/meetings/ListThemesEndpointTest.java b/src/test/java/com/vonage/client/meetings/ListThemesEndpointTest.java
index 85841f7..1446563 100644
--- a/src/test/java/com/vonage/client/meetings/ListThemesEndpointTest.java
+++ b/src/test/java/com/vonage/client/meetings/ListThemesEndpointTest.java
@@ -40,7 +40,7 @@ public class ListThemesEndpointTest {
 		Void request = null;
 		RequestBuilder builder = endpoint.makeRequest(request);
 		assertEquals("GET", builder.getMethod());
-		String expectedUri = "https://api-eu.vonage.com/meetings/themes";
+		String expectedUri = "https://api-eu.vonage.com/v1/meetings/themes";
 		assertEquals(expectedUri, builder.build().getURI().toString());
 		HttpResponse mockResponse = TestUtils.makeJsonHttpResponse(200, "[{}]");
 		List<Theme> parsed = endpoint.parseResponse(mockResponse);
@@ -53,7 +53,7 @@ public class ListThemesEndpointTest {
 		String baseUri = "http://example.com";
 		HttpWrapper wrapper = new HttpWrapper(HttpConfig.builder().baseUri(baseUri).build());
 		endpoint = new ListThemesEndpoint(wrapper);
-		String expectedUri = baseUri + "/meetings/themes";
+		String expectedUri = baseUri + "/v1/meetings/themes";
 		RequestBuilder builder = endpoint.makeRequest(null);
 		assertEquals(expectedUri, builder.build().getURI().toString());
 		assertEquals("GET", builder.getMethod());
diff --git a/src/test/java/com/vonage/client/meetings/SearchThemeRoomsEndpointTest.java b/src/test/java/com/vonage/client/meetings/SearchThemeRoomsEndpointTest.java
index ef86fde..929a948 100644
--- a/src/test/java/com/vonage/client/meetings/SearchThemeRoomsEndpointTest.java
+++ b/src/test/java/com/vonage/client/meetings/SearchThemeRoomsEndpointTest.java
@@ -42,7 +42,7 @@ public class SearchThemeRoomsEndpointTest {
 		ListRoomsRequest request = new ListRoomsRequest(21, 33, 50, UUID.randomUUID());
 		RequestBuilder builder = endpoint.makeRequest(request);
 		assertEquals("GET", builder.getMethod());
-		String expectedUri = "https://api-eu.vonage.com/meetings/themes/"+request.themeId+
+		String expectedUri = "https://api-eu.vonage.com/v1/meetings/themes/"+request.themeId+
 				"/rooms?start_id="+request.startId+"&end_id="+request.endId+"&page_size="+request.pageSize;
 		assertEquals(expectedUri, builder.build().getURI().toString());
 		Map<String, String> params = TestUtils.makeParameterMap(builder.getParameters());
@@ -70,7 +70,7 @@ public class SearchThemeRoomsEndpointTest {
 		HttpWrapper wrapper = new HttpWrapper(HttpConfig.builder().baseUri(baseUri).build());
 		endpoint = new SearchThemeRoomsEndpoint(wrapper);
 		ListRoomsRequest request = new ListRoomsRequest(1, null, null, UUID.randomUUID());
-		String expectedUri = baseUri + "/meetings/themes/"+request.themeId+"/rooms?start_id="+request.startId;
+		String expectedUri = baseUri + "/v1/meetings/themes/"+request.themeId+"/rooms?start_id="+request.startId;
 		RequestBuilder builder = endpoint.makeRequest(request);
 		assertEquals(expectedUri, builder.build().getURI().toString());
 		Map<String, String> params = TestUtils.makeParameterMap(builder.getParameters());
diff --git a/src/test/java/com/vonage/client/meetings/UpdateApplicationEndpointTest.java b/src/test/java/com/vonage/client/meetings/UpdateApplicationEndpointTest.java
index 971c10a..a9ffa07 100644
--- a/src/test/java/com/vonage/client/meetings/UpdateApplicationEndpointTest.java
+++ b/src/test/java/com/vonage/client/meetings/UpdateApplicationEndpointTest.java
@@ -43,7 +43,7 @@ public class UpdateApplicationEndpointTest {
 			.build();
 		RequestBuilder builder = endpoint.makeRequest(request);
 		assertEquals("PATCH", builder.getMethod());
-		String expectedUri = "https://api-eu.vonage.com/meetings/applications";
+		String expectedUri = "https://api-eu.vonage.com/v1/meetings/applications";
 		assertEquals(expectedUri, builder.build().getURI().toString());
 		assertEquals(ContentType.APPLICATION_JSON.getMimeType(), builder.getFirstHeader("Content-Type").getValue());
 		assertEquals(ContentType.APPLICATION_JSON.getMimeType(), builder.getFirstHeader("Accept").getValue());
@@ -66,7 +66,7 @@ public class UpdateApplicationEndpointTest {
 		String baseUri = "http://example.com";
 		HttpWrapper wrapper = new HttpWrapper(HttpConfig.builder().baseUri(baseUri).build());
 		endpoint = new UpdateApplicationEndpoint(wrapper);
-		String expectedUri = baseUri + "/meetings/applications";
+		String expectedUri = baseUri + "/v1/meetings/applications";
 		UpdateApplicationRequest request = UpdateApplicationRequest.builder().defaultThemeId(UUID.randomUUID()).build();
 		RequestBuilder builder = endpoint.makeRequest(request);
 		assertEquals(expectedUri, builder.build().getURI().toString());
diff --git a/src/test/java/com/vonage/client/meetings/UpdateRoomEndpointTest.java b/src/test/java/com/vonage/client/meetings/UpdateRoomEndpointTest.java
index 6bd9b4d..ef37910 100644
--- a/src/test/java/com/vonage/client/meetings/UpdateRoomEndpointTest.java
+++ b/src/test/java/com/vonage/client/meetings/UpdateRoomEndpointTest.java
@@ -55,7 +55,7 @@ public class UpdateRoomEndpointTest {
 		request.roomId = roomId;
 		RequestBuilder builder = endpoint.makeRequest(request);
 		assertEquals("PATCH", builder.getMethod());
-		String expectedUri = "https://api-eu.vonage.com/meetings/rooms/"+roomId;
+		String expectedUri = "https://api-eu.vonage.com/v1/meetings/rooms/"+roomId;
 		assertEquals(expectedUri, builder.build().getURI().toString());
 
 		assertEquals(ContentType.APPLICATION_JSON.getMimeType(), builder.getFirstHeader("Content-Type").getValue());
@@ -73,7 +73,7 @@ public class UpdateRoomEndpointTest {
 		String baseUri = "http://example.com";
 		HttpWrapper wrapper = new HttpWrapper(HttpConfig.builder().baseUri(baseUri).build());
 		endpoint = new UpdateRoomEndpoint(wrapper);
-		String expectedUri = baseUri + "/meetings/rooms/"+MeetingsClientTest.RANDOM_ID;
+		String expectedUri = baseUri + "/v1/meetings/rooms/"+MeetingsClientTest.RANDOM_ID;
 		UpdateRoomRequest request = UpdateRoomRequest.builder().build();
 		request.roomId = MeetingsClientTest.RANDOM_ID;
 		RequestBuilder builder = endpoint.makeRequest(request);
diff --git a/src/test/java/com/vonage/client/meetings/UpdateThemeEndpointTest.java b/src/test/java/com/vonage/client/meetings/UpdateThemeEndpointTest.java
index 330c8dc..5644bed 100644
--- a/src/test/java/com/vonage/client/meetings/UpdateThemeEndpointTest.java
+++ b/src/test/java/com/vonage/client/meetings/UpdateThemeEndpointTest.java
@@ -45,7 +45,7 @@ public class UpdateThemeEndpointTest {
 		request.themeId = UUID.fromString(themeId);
 		RequestBuilder builder = endpoint.makeRequest(request);
 		assertEquals("PATCH", builder.getMethod());
-		String expectedUri = "https://api-eu.vonage.com/meetings/themes/"+themeId;
+		String expectedUri = "https://api-eu.vonage.com/v1/meetings/themes/"+themeId;
 		assertEquals(expectedUri, builder.build().getURI().toString());
 
 		assertEquals(ContentType.APPLICATION_JSON.getMimeType(), builder.getFirstHeader("Content-Type").getValue());
@@ -67,7 +67,7 @@ public class UpdateThemeEndpointTest {
 		endpoint = new UpdateThemeEndpoint(wrapper);
 		Theme request = Theme.builder().build();
 		request.themeId = UUID.randomUUID();
-		String expectedUri = baseUri + "/meetings/themes/"+request.themeId;
+		String expectedUri = baseUri + "/v1/meetings/themes/"+request.themeId;
 		RequestBuilder builder = endpoint.makeRequest(request);
 		assertEquals(expectedUri, builder.build().getURI().toString());
 		assertEquals(ContentType.APPLICATION_JSON.getMimeType(), builder.getFirstHeader("Content-Type").getValue());
