diff --git a/src/main/java/com/headius/invokebinder/transform/Varargs.java b/src/main/java/com/headius/invokebinder/transform/Varargs.java
index 6c2d099..b4f34a4 100644
--- a/src/main/java/com/headius/invokebinder/transform/Varargs.java
+++ b/src/main/java/com/headius/invokebinder/transform/Varargs.java
@@ -51,7 +51,7 @@ public class Varargs extends Transform {
         Class<?> componentType = arrayType.getComponentType();
         for (int i = index; i < source.parameterCount(); i++) {
             Class<?> in = source.parameterType(i);
-            assert in.isAssignableFrom(componentType)
+            assert componentType.isAssignableFrom(in)
                     : "incoming type " + in.getName() + " not compatible with " + componentType.getName() + "[]";
         }
     }
