diff --git a/src/test/java/com/fincatto/documentofiscal/cte400/classes/CTAutorizador400Test.java b/src/test/java/com/fincatto/documentofiscal/cte400/classes/CTAutorizador400Test.java
index 7c8c64a..46c0fb8 100644
--- a/src/test/java/com/fincatto/documentofiscal/cte400/classes/CTAutorizador400Test.java
+++ b/src/test/java/com/fincatto/documentofiscal/cte400/classes/CTAutorizador400Test.java
@@ -17,12 +17,12 @@ public class CTAutorizador400Test {
     	Assert.assertEquals("https://homologacao.sefaz.mt.gov.br/ctews2/services/CTeStatusServicoV4", autorizador.getCteStatusServico(DFAmbiente.HOMOLOGACAO));
     	Assert.assertEquals("https://homologacao.sefaz.mt.gov.br/ctews2/services/CTeRecepcaoEventoV4", autorizador.getRecepcaoEvento(DFAmbiente.HOMOLOGACAO));
 
-		Assert.assertNull(autorizador.getCteRecepcaoSinc(DFAmbiente.PRODUCAO));
-		Assert.assertNull(autorizador.getCteRecepcaoOS(DFAmbiente.PRODUCAO));
-		Assert.assertNull(autorizador.getCteRecepcaoGTVe(DFAmbiente.PRODUCAO));
-		Assert.assertNull(autorizador.getCteConsultaProtocolo(DFAmbiente.PRODUCAO));
-		Assert.assertNull(autorizador.getCteStatusServico(DFAmbiente.PRODUCAO));
-		Assert.assertNull(autorizador.getRecepcaoEvento(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://cte.sefaz.mt.gov.br/ctews2/services/CTeRecepcaoSincV4", autorizador.getCteRecepcaoSinc(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://cte.sefaz.mt.gov.br/ctews/services/CTeRecepcaoOSV4", autorizador.getCteRecepcaoOS(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://cte.sefaz.mt.gov.br/ctews2/services/CTeRecepcaoGTVeV4", autorizador.getCteRecepcaoGTVe(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://cte.sefaz.mt.gov.br/ctews2/services/CTeConsultaV4", autorizador.getCteConsultaProtocolo(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://cte.sefaz.mt.gov.br/ctews2/services/CTeStatusServicoV4", autorizador.getCteStatusServico(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://cte.sefaz.mt.gov.br/ctews2/services/CTeRecepcaoEventoV4", autorizador.getRecepcaoEvento(DFAmbiente.PRODUCAO));
     }
 
     @Test
@@ -36,12 +36,12 @@ public class CTAutorizador400Test {
     	Assert.assertEquals("https://homologacao.cte.ms.gov.br/ws/CTeStatusServicoV4", autorizador.getCteStatusServico(DFAmbiente.HOMOLOGACAO));
     	Assert.assertEquals("https://homologacao.cte.ms.gov.br/ws/CTeRecepcaoEventoV4", autorizador.getRecepcaoEvento(DFAmbiente.HOMOLOGACAO));
 
-		Assert.assertNull(autorizador.getCteRecepcaoSinc(DFAmbiente.PRODUCAO));
-		Assert.assertNull(autorizador.getCteRecepcaoOS(DFAmbiente.PRODUCAO));
-		Assert.assertNull(autorizador.getCteRecepcaoGTVe(DFAmbiente.PRODUCAO));
-		Assert.assertNull(autorizador.getCteConsultaProtocolo(DFAmbiente.PRODUCAO));
-		Assert.assertNull(autorizador.getCteStatusServico(DFAmbiente.PRODUCAO));
-		Assert.assertNull(autorizador.getRecepcaoEvento(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://producao.cte.ms.gov.br/ws/CTeRecepcaoSincV4", autorizador.getCteRecepcaoSinc(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://producao.cte.ms.gov.br/ws/CTeRecepcaoOSV4", autorizador.getCteRecepcaoOS(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://producao.cte.ms.gov.br/ws/CTeRecepcaoGTVeV4", autorizador.getCteRecepcaoGTVe(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://producao.cte.ms.gov.br/ws/CTeConsultaV4", autorizador.getCteConsultaProtocolo(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://producao.cte.ms.gov.br/ws/CTeStatusServicoV4", autorizador.getCteStatusServico(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://producao.cte.ms.gov.br/ws/CTeRecepcaoEventoV4", autorizador.getRecepcaoEvento(DFAmbiente.PRODUCAO));
     }
     
     @Test
@@ -55,12 +55,12 @@ public class CTAutorizador400Test {
     	Assert.assertEquals("https://hcte.fazenda.mg.gov.br/cte/services/CTeStatusServicoV4", autorizador.getCteStatusServico(DFAmbiente.HOMOLOGACAO));
     	Assert.assertEquals("https://hcte.fazenda.mg.gov.br/cte/services/CTeRecepcaoEventoV4", autorizador.getRecepcaoEvento(DFAmbiente.HOMOLOGACAO));
 
-		Assert.assertNull(autorizador.getCteRecepcaoSinc(DFAmbiente.PRODUCAO));
-		Assert.assertNull(autorizador.getCteRecepcaoOS(DFAmbiente.PRODUCAO));
-		Assert.assertNull(autorizador.getCteRecepcaoGTVe(DFAmbiente.PRODUCAO));
-		Assert.assertNull(autorizador.getCteConsultaProtocolo(DFAmbiente.PRODUCAO));
-		Assert.assertNull(autorizador.getCteStatusServico(DFAmbiente.PRODUCAO));
-		Assert.assertNull(autorizador.getRecepcaoEvento(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://cte.fazenda.mg.gov.br/cte/services/CTeRecepcaoSincV4", autorizador.getCteRecepcaoSinc(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://cte.fazenda.mg.gov.br/cte/services/CTeRecepcaoOSV4", autorizador.getCteRecepcaoOS(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://cte.fazenda.mg.gov.br/cte/services/CTeRecepcaoGTVeV4", autorizador.getCteRecepcaoGTVe(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://cte.fazenda.mg.gov.br/cte/services/CTeConsultaV4", autorizador.getCteConsultaProtocolo(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://cte.fazenda.mg.gov.br/cte/services/CTeStatusServicoV4", autorizador.getCteStatusServico(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://cte.fazenda.mg.gov.br/cte/services/CTeRecepcaoEventoV4", autorizador.getRecepcaoEvento(DFAmbiente.PRODUCAO));
     }
     
     @Test
@@ -73,13 +73,13 @@ public class CTAutorizador400Test {
     	Assert.assertEquals("https://homologacao.cte.fazenda.pr.gov.br/cte4/CTeConsultaV4", autorizador.getCteConsultaProtocolo(DFAmbiente.HOMOLOGACAO));
     	Assert.assertEquals("https://homologacao.cte.fazenda.pr.gov.br/cte4/CTeStatusServicoV4", autorizador.getCteStatusServico(DFAmbiente.HOMOLOGACAO));
     	Assert.assertEquals("https://homologacao.cte.fazenda.pr.gov.br/cte4/CTeRecepcaoEventoV4", autorizador.getRecepcaoEvento(DFAmbiente.HOMOLOGACAO));
-    	
-    	Assert.assertNull(autorizador.getCteRecepcaoSinc(DFAmbiente.PRODUCAO));
-		Assert.assertNull(autorizador.getCteRecepcaoOS(DFAmbiente.PRODUCAO));
-		Assert.assertNull(autorizador.getCteRecepcaoGTVe(DFAmbiente.PRODUCAO));
-    	Assert.assertNull(autorizador.getCteConsultaProtocolo(DFAmbiente.PRODUCAO));
-    	Assert.assertNull(autorizador.getCteStatusServico(DFAmbiente.PRODUCAO));
-    	Assert.assertNull(autorizador.getRecepcaoEvento(DFAmbiente.PRODUCAO));
+
+		Assert.assertEquals("https://cte.fazenda.pr.gov.br/cte4/CTeRecepcaoSincV4", autorizador.getCteRecepcaoSinc(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://cte.fazenda.pr.gov.br/cte4/CTeRecepcaoOSV4", autorizador.getCteRecepcaoOS(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://cte.fazenda.pr.gov.br/cte4/CTeRecepcaoGTVeV4", autorizador.getCteRecepcaoGTVe(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://cte.fazenda.pr.gov.br/cte4/CTeConsultaV4", autorizador.getCteConsultaProtocolo(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://cte.fazenda.pr.gov.br/cte4/CTeStatusServicoV4", autorizador.getCteStatusServico(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://cte.fazenda.pr.gov.br/cte4/CTeRecepcaoEventoV4", autorizador.getRecepcaoEvento(DFAmbiente.PRODUCAO));
     }
     
     @Test
@@ -112,12 +112,12 @@ public class CTAutorizador400Test {
 		Assert.assertEquals("https://homologacao.nfe.fazenda.sp.gov.br/CTeWS/WS/CTeStatusServicoV4.asmx", autorizador.getCteStatusServico(DFAmbiente.HOMOLOGACAO));
 		Assert.assertEquals("https://homologacao.nfe.fazenda.sp.gov.br/CTeWS/WS/CTeRecepcaoEventoV4.asmx", autorizador.getRecepcaoEvento(DFAmbiente.HOMOLOGACAO));
 
-		Assert.assertNull(autorizador.getCteRecepcaoSinc(DFAmbiente.PRODUCAO));
-		Assert.assertNull(autorizador.getCteRecepcaoOS(DFAmbiente.PRODUCAO));
-		Assert.assertNull(autorizador.getCteRecepcaoGTVe(DFAmbiente.PRODUCAO));
-		Assert.assertNull(autorizador.getCteConsultaProtocolo(DFAmbiente.PRODUCAO));
-		Assert.assertNull(autorizador.getCteStatusServico(DFAmbiente.PRODUCAO));
-		Assert.assertNull(autorizador.getRecepcaoEvento(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://nfe.fazenda.sp.gov.br/CTeWS/WS/CTeRecepcaoSincV4.asmx", autorizador.getCteRecepcaoSinc(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://nfe.fazenda.sp.gov.br/CTeWS/WS/CTeRecepcaoOSV4.asmx", autorizador.getCteRecepcaoOS(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://nfe.fazenda.sp.gov.br/CTeWS/WS/CTeRecepcaoGTVeV4.asmx", autorizador.getCteRecepcaoGTVe(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://nfe.fazenda.sp.gov.br/CTeWS/WS/CTeConsultaV4.asmx", autorizador.getCteConsultaProtocolo(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://nfe.fazenda.sp.gov.br/CTeWS/WS/CTeStatusServicoV4.asmx", autorizador.getCteStatusServico(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://nfe.fazenda.sp.gov.br/CTeWS/WS/CTeRecepcaoEventoV4.asmx", autorizador.getRecepcaoEvento(DFAmbiente.PRODUCAO));
     }
     
     @Test
@@ -149,12 +149,12 @@ public class CTAutorizador400Test {
     	Assert.assertEquals("https://homologacao.nfe.fazenda.sp.gov.br/CTeWS/WS/CTeConsultaV4.asmx", autorizador.getCteConsultaProtocolo(DFAmbiente.HOMOLOGACAO));
     	Assert.assertEquals("https://homologacao.nfe.fazenda.sp.gov.br/CTeWS/WS/CTeStatusServicoV4.asmx", autorizador.getCteStatusServico(DFAmbiente.HOMOLOGACAO));
 		Assert.assertEquals("https://homologacao.nfe.fazenda.sp.gov.br/CTeWS/WS/CTeRecepcaoEventoV4.asmx", autorizador.getRecepcaoEvento(DFAmbiente.HOMOLOGACAO));
-    	
-    	Assert.assertNull(autorizador.getCteRecepcaoSinc(DFAmbiente.PRODUCAO));
-		Assert.assertNull(autorizador.getCteRecepcaoOS(DFAmbiente.PRODUCAO));
-		Assert.assertNull(autorizador.getCteRecepcaoGTVe(DFAmbiente.PRODUCAO));
-    	Assert.assertNull(autorizador.getCteConsultaProtocolo(DFAmbiente.PRODUCAO));
-    	Assert.assertNull(autorizador.getCteStatusServico(DFAmbiente.PRODUCAO));
-    	Assert.assertNull(autorizador.getRecepcaoEvento(DFAmbiente.PRODUCAO));
+
+		Assert.assertEquals("https://nfe.fazenda.sp.gov.br/CTeWS/WS/CTeRecepcaoSincV4.asmx", autorizador.getCteRecepcaoSinc(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://nfe.fazenda.sp.gov.br/CTeWS/WS/CTeRecepcaoOSV4.asmx", autorizador.getCteRecepcaoOS(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://nfe.fazenda.sp.gov.br/CTeWS/WS/CTeRecepcaoGTVeV4.asmx", autorizador.getCteRecepcaoGTVe(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://nfe.fazenda.sp.gov.br/CTeWS/WS/CTeConsultaV4.asmx", autorizador.getCteConsultaProtocolo(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://nfe.fazenda.sp.gov.br/CTeWS/WS/CTeStatusServicoV4.asmx", autorizador.getCteStatusServico(DFAmbiente.PRODUCAO));
+		Assert.assertEquals("https://nfe.fazenda.sp.gov.br/CTeWS/WS/CTeRecepcaoEventoV4.asmx", autorizador.getRecepcaoEvento(DFAmbiente.PRODUCAO));
     }
 }
