diff --git a/src/test/resources/Formatted Strings.txt b/src/test/resources/Formatted Strings.txt
index dfe97b4..cbde82d 100644
--- a/src/test/resources/Formatted Strings.txt
+++ b/src/test/resources/Formatted Strings.txt
@@ -942,4 +942,20 @@ builder.append(current);
                return numberOfCharacters % 2 == 1;
                       }
                          }
+----
+/*/Should still be inside a multi line comment {{
+same indentation
+end of comment */
+----
+/*/Should still be inside a multi line comment {{
+same indentation
+end of comment */
+----
+/*/Should still be inside a multi line comment {{
+same indentation
+end of comment */
+----
+        /*/Should still be inside a multi line comment {{
+                same indentation
+     end of comment */
 ----
\ No newline at end of file
diff --git a/src/test/resources/Unformatted Strings.txt b/src/test/resources/Unformatted Strings.txt
index bce8000..7730f07 100644
--- a/src/test/resources/Unformatted Strings.txt
+++ b/src/test/resources/Unformatted Strings.txt
@@ -234,3 +234,8 @@ builder.append(current);
                return numberOfCharacters % 2 == 1;
                       }
                          }
+----
+        /*/Should still be inside a multi line comment {{
+                same indentation
+     end of comment */
+----
\ No newline at end of file
