diff --git a/src/main/java/com/vonage/client/application/Application.java b/src/main/java/com/vonage/client/application/Application.java
index 18a2a54..e68b288 100644
--- a/src/main/java/com/vonage/client/application/Application.java
+++ b/src/main/java/com/vonage/client/application/Application.java
@@ -117,7 +117,7 @@ public class Application {
          * @return The {@link Builder} to keep building.
          */
         public Builder name(String name) {
-            name = name;
+            this.name = name;
             return this;
         }
 
