diff --git a/src/main/java/com/google/code/externalsorting/ExternalSort.java b/src/main/java/com/google/code/externalsorting/ExternalSort.java
index e907b6d..99ab6c2 100644
--- a/src/main/java/com/google/code/externalsorting/ExternalSort.java
+++ b/src/main/java/com/google/code/externalsorting/ExternalSort.java
@@ -447,7 +447,7 @@ public class ExternalSort {
          */
         public static void sort(final File input, final File output, final Comparator<String> cmp)
                 throws IOException {
-                ExternalSort.mergeSortedFiles(ExternalSort.sortInBatch(input),
+                ExternalSort.mergeSortedFiles(ExternalSort.sortInBatch(input, cmp),
                         output, cmp);
         }
 
