diff --git a/src/test/java/org/traccar/protocol/T622IridiumProtocolDecoderTest.java b/src/test/java/org/traccar/protocol/T622IridiumProtocolDecoderTest.java
index 17b2526..ff5d5b0 100644
--- a/src/test/java/org/traccar/protocol/T622IridiumProtocolDecoderTest.java
+++ b/src/test/java/org/traccar/protocol/T622IridiumProtocolDecoderTest.java
@@ -1,12 +1,10 @@
 package org.traccar.protocol;
 
-import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 import org.traccar.ProtocolTest;
 
 public class T622IridiumProtocolDecoderTest extends ProtocolTest {
 
-    @Disabled
     @Test
     public void testDecode() throws Exception {
 
@@ -15,7 +13,8 @@ public class T622IridiumProtocolDecoderTest extends ProtocolTest {
         decoder.setFormat("01,02,03,04,05,08");
 
         verifyPosition(decoder, binary(
-                "01003301001c2a8cef8333303034333430363735343836353000001700006461d512020011232f03a0fff1c85d0612b3f02b00000048"));
+                "01003501001c68b2cb1733303034333430363735343836353000016e000064b5f497020013234c5ea0ff1c365d0600b1482c010000cf0004"),
+                position("2023-07-18 02:10:08.000", true, -6.26732, 106.77200));
 
     }
 
