diff --git a/src/test/java/com/fincatto/documentofiscal/nfe400/classes/NFNotaInfoTipoVeiculoTest.java b/src/test/java/com/fincatto/documentofiscal/nfe400/classes/NFNotaInfoTipoVeiculoTest.java
index 264f804..132a1f1 100755
--- a/src/test/java/com/fincatto/documentofiscal/nfe400/classes/NFNotaInfoTipoVeiculoTest.java
+++ b/src/test/java/com/fincatto/documentofiscal/nfe400/classes/NFNotaInfoTipoVeiculoTest.java
@@ -7,26 +7,31 @@ public class NFNotaInfoTipoVeiculoTest {
     
     @Test
     public void deveRepresentarOCodigoCorretamente() {
-        Assert.assertEquals("06", NFNotaInfoTipoVeiculo.AUTOMOVEL.getCodigo());
-        Assert.assertEquals("14", NFNotaInfoTipoVeiculo.CAMINHAO.getCodigo());
-        Assert.assertEquals("13", NFNotaInfoTipoVeiculo.CAMINHONETA.getCodigo());
-        Assert.assertEquals("24", NFNotaInfoTipoVeiculo.CARGA_CAM.getCodigo());
-        Assert.assertEquals("02", NFNotaInfoTipoVeiculo.CICLOMOTO.getCodigo());
-        Assert.assertEquals("22", NFNotaInfoTipoVeiculo.ESP_ONIBUS.getCodigo());
-        Assert.assertEquals("07", NFNotaInfoTipoVeiculo.MICROONIBUS.getCodigo());
-        Assert.assertEquals("23", NFNotaInfoTipoVeiculo.MISTO_CAM.getCodigo());
-        Assert.assertEquals("04", NFNotaInfoTipoVeiculo.MOTOCICLO.getCodigo());
-        Assert.assertEquals("03", NFNotaInfoTipoVeiculo.MOTONETA.getCodigo());
-        Assert.assertEquals("08", NFNotaInfoTipoVeiculo.ONIBUS.getCodigo());
+        Assert.assertEquals("2", NFNotaInfoTipoVeiculo.CICLOMOTO.getCodigo());
+        Assert.assertEquals("3", NFNotaInfoTipoVeiculo.MOTONETA.getCodigo());
+        Assert.assertEquals("4", NFNotaInfoTipoVeiculo.MOTOCICLO.getCodigo());
+        Assert.assertEquals("5", NFNotaInfoTipoVeiculo.TRICICLO.getCodigo());
+        Assert.assertEquals("6", NFNotaInfoTipoVeiculo.AUTOMOVEL.getCodigo());
+        Assert.assertEquals("7", NFNotaInfoTipoVeiculo.MICROONIBUS.getCodigo());
+        Assert.assertEquals("8", NFNotaInfoTipoVeiculo.ONIBUS.getCodigo());
         Assert.assertEquals("10", NFNotaInfoTipoVeiculo.REBOQUE.getCodigo());
-        Assert.assertEquals("05", NFNotaInfoTipoVeiculo.TRICICLO.getCodigo());
-        Assert.assertEquals("17", NFNotaInfoTipoVeiculo.TRATOR.getCodigo());
         Assert.assertEquals("11", NFNotaInfoTipoVeiculo.SEMIRREBOQUE.getCodigo());
+        Assert.assertEquals("13", NFNotaInfoTipoVeiculo.CAMINHONETA.getCodigo());
+        Assert.assertEquals("14", NFNotaInfoTipoVeiculo.CAMINHAO.getCodigo());
+        Assert.assertEquals("17", NFNotaInfoTipoVeiculo.TRATOR.getCodigo());
         Assert.assertEquals("18", NFNotaInfoTipoVeiculo.TRATOR_RODAS.getCodigo());
         Assert.assertEquals("19", NFNotaInfoTipoVeiculo.TRATOR_ESTEIRAS.getCodigo());
         Assert.assertEquals("20", NFNotaInfoTipoVeiculo.TRATOR_MISTO.getCodigo());
         Assert.assertEquals("21", NFNotaInfoTipoVeiculo.QUADRICICLO.getCodigo());
+        Assert.assertEquals("22", NFNotaInfoTipoVeiculo.ESP_ONIBUS.getCodigo());
+        Assert.assertEquals("23", NFNotaInfoTipoVeiculo.MISTO_CAM.getCodigo());
+        Assert.assertEquals("24", NFNotaInfoTipoVeiculo.CARGA_CAM.getCodigo());
         Assert.assertEquals("25", NFNotaInfoTipoVeiculo.UTILITARIO.getCodigo());
         Assert.assertEquals("26", NFNotaInfoTipoVeiculo.MOTOR_CASA.getCodigo());
+
+
+
+
+
     }
 }
\ No newline at end of file
diff --git a/src/test/java/com/fincatto/documentofiscal/nfe400/classes/nota/NFNotaInfoItemProdutoVeiculoTest.java b/src/test/java/com/fincatto/documentofiscal/nfe400/classes/nota/NFNotaInfoItemProdutoVeiculoTest.java
index 9173021..4c07800 100755
--- a/src/test/java/com/fincatto/documentofiscal/nfe400/classes/nota/NFNotaInfoItemProdutoVeiculoTest.java
+++ b/src/test/java/com/fincatto/documentofiscal/nfe400/classes/nota/NFNotaInfoItemProdutoVeiculoTest.java
@@ -838,7 +838,7 @@ public class NFNotaInfoItemProdutoVeiculoTest {
 
     @Test
     public void deveGerarXMLDeAcordoComOPadraoEstabelecido() {
-        final String xmlEsperado = "<NFNotaInfoItemProdutoVeiculo><tpOp>0</tpOp><chassi>B31YTE9INXEYDMO4C</chassi><cCor>123A</cCor><xCor>tjRQlvBqVsFLDWUCaOeaXRbV1ipOZH3W5H7Ic2p9</xCor><pot>GZRz</pot><cilin>ohNx</cilin><pesoL>9999.9999</pesoL><pesoB>9999.9999</pesoB><nSerie>XBxVyiVVD</nSerie><tpComb>02</tpComb><nMotor>u1DU8GZRzJZzevuY49IB9</nMotor><CMT>9999.9999</CMT><dist>9999</dist><anoMod>2014</anoMod><anoFab>2014</anoFab><tpPint>A</tpPint><tpVeic>06</tpVeic><espVeic>4</espVeic><VIN>N</VIN><condVeic>1</condVeic><cMod>542562</cMod><cCorDENATRAN>16</cCorDENATRAN><lota>999</lota><tpRest>4</tpRest></NFNotaInfoItemProdutoVeiculo>";
+        final String xmlEsperado = "<NFNotaInfoItemProdutoVeiculo><tpOp>0</tpOp><chassi>B31YTE9INXEYDMO4C</chassi><cCor>123A</cCor><xCor>tjRQlvBqVsFLDWUCaOeaXRbV1ipOZH3W5H7Ic2p9</xCor><pot>GZRz</pot><cilin>ohNx</cilin><pesoL>9999.9999</pesoL><pesoB>9999.9999</pesoB><nSerie>XBxVyiVVD</nSerie><tpComb>02</tpComb><nMotor>u1DU8GZRzJZzevuY49IB9</nMotor><CMT>9999.9999</CMT><dist>9999</dist><anoMod>2014</anoMod><anoFab>2014</anoFab><tpPint>A</tpPint><tpVeic>6</tpVeic><espVeic>4</espVeic><VIN>N</VIN><condVeic>1</condVeic><cMod>542562</cMod><cCorDENATRAN>16</cCorDENATRAN><lota>999</lota><tpRest>4</tpRest></NFNotaInfoItemProdutoVeiculo>";
         Assert.assertEquals(xmlEsperado, FabricaDeObjetosFake.getNFNotaInfoItemProdutoVeiculo().toString());
     }
 }
\ No newline at end of file
