diff --git a/src/test/java/io/kokuwa/maven/helm/AbstractHelmMojoTest.java b/src/test/java/io/kokuwa/maven/helm/AbstractHelmMojoTest.java
index 0eed091..3b477cb 100644
--- a/src/test/java/io/kokuwa/maven/helm/AbstractHelmMojoTest.java
+++ b/src/test/java/io/kokuwa/maven/helm/AbstractHelmMojoTest.java
@@ -98,7 +98,7 @@ class AbstractHelmMojoTest {
 			K8SCluster k8sCluster = new K8SCluster();
 			subjectSpy.setK8sCluster(k8sCluster);
 			k8sCluster.setApiUrl("custom-api-url");
-			assertEquals(" --kube-apiserver custom-api-url", subjectSpy.getK8SArgs());
+			assertEquals(" --kube-apiserver=custom-api-url", subjectSpy.getK8SArgs());
 		}
 
 		@Test
@@ -106,7 +106,7 @@ class AbstractHelmMojoTest {
 			K8SCluster k8sCluster = new K8SCluster();
 			subjectSpy.setK8sCluster(k8sCluster);
 			k8sCluster.setAsUser("custom-user");
-			assertEquals(" --kube-as-user custom-user", subjectSpy.getK8SArgs());
+			assertEquals(" --kube-as-user=custom-user", subjectSpy.getK8SArgs());
 		}
 
 		@Test
@@ -114,23 +114,16 @@ class AbstractHelmMojoTest {
 			K8SCluster k8sCluster = new K8SCluster();
 			subjectSpy.setK8sCluster(k8sCluster);
 			k8sCluster.setAsGroup("custom-group");
-			assertEquals(" --kube-as-group custom-group", subjectSpy.getK8SArgs());
+			assertEquals(" --kube-as-group=custom-group", subjectSpy.getK8SArgs());
 		}
 
-		@Test
-		void k8sClusterArg_Namespace() {
-			K8SCluster k8sCluster = new K8SCluster();
-			subjectSpy.setK8sCluster(k8sCluster);
-			k8sCluster.setNamespace("custom-ns");
-			assertEquals(" --namespace custom-ns", subjectSpy.getK8SArgs());
-		}
 
 		@Test
 		void k8sClusterArg_Token() {
 			K8SCluster k8sCluster = new K8SCluster();
 			subjectSpy.setK8sCluster(k8sCluster);
 			k8sCluster.setToken("custom-token");
-			assertEquals(" --kube-token custom-token", subjectSpy.getK8SArgs());
+			assertEquals(" --kube-token=custom-token", subjectSpy.getK8SArgs());
 		}
 
 		@Test
@@ -141,13 +134,11 @@ class AbstractHelmMojoTest {
 			k8sCluster.setAsUser("custom-user");
 			k8sCluster.setAsGroup("custom-group");
 			k8sCluster.setToken("custom-token");
-			k8sCluster.setNamespace("custom-ns");
 			k8sCluster.setToken("custom-token");
-			assertEquals(" --kube-apiserver custom-api-url"
-					+ " --namespace custom-ns"
-					+ " --kube-as-user custom-user"
-					+ " --kube-as-group custom-group"
-					+ " --kube-token custom-token",
+			assertEquals(" --kube-apiserver=custom-api-url"
+					+ " --kube-as-user=custom-user"
+					+ " --kube-as-group=custom-group"
+					+ " --kube-token=custom-token",
 					subjectSpy.getK8SArgs());
 		}
 	}
diff --git a/src/test/java/io/kokuwa/maven/helm/HideSensitiveDataLogTest.java b/src/test/java/io/kokuwa/maven/helm/HideSensitiveDataLogTest.java
index b557533..aba9fe1 100644
--- a/src/test/java/io/kokuwa/maven/helm/HideSensitiveDataLogTest.java
+++ b/src/test/java/io/kokuwa/maven/helm/HideSensitiveDataLogTest.java
@@ -29,8 +29,8 @@ public class HideSensitiveDataLogTest {
 
 	@Test
 	void hideKubeToken() {
-		String content = "abcd --kube-token ABCDEF";
-		String expected = "abcd --kube-token *****";
+		String content = "abcd --kube-token=ABCDEF";
+		String expected = "abcd --kube-token=*****";
 		String actual = StripSensitiveDataLog.strip(content);
 		assertEquals(expected, actual);
 	}
