diff --git a/src/test/java/com/googlecode/aviator/test/function/GrammarUnitTest.java b/src/test/java/com/googlecode/aviator/test/function/GrammarUnitTest.java
index 4bab314..32196dc 100644
--- a/src/test/java/com/googlecode/aviator/test/function/GrammarUnitTest.java
+++ b/src/test/java/com/googlecode/aviator/test/function/GrammarUnitTest.java
@@ -31,6 +31,7 @@ import java.util.Map;
 import org.junit.Test;
 import com.googlecode.aviator.AviatorEvaluator;
 import com.googlecode.aviator.AviatorEvaluatorInstance;
+import com.googlecode.aviator.Expression;
 import com.googlecode.aviator.Options;
 import com.googlecode.aviator.exception.CompileExpressionErrorException;
 import com.googlecode.aviator.exception.ExpressionRuntimeException;
@@ -49,6 +50,22 @@ import junit.framework.Assert;
  */
 public class GrammarUnitTest {
 
+
+  @Test
+  public void testIssue186() {
+    Expression exp = AviatorEvaluator.compile("1==1");
+    assertTrue(exp.getVariableFullNames().isEmpty());
+    assertTrue(exp.getVariableNames().isEmpty());
+
+    exp = AviatorEvaluator.compile("a!=nil");
+    List<String> vars = exp.getVariableFullNames();
+    assertEquals(1, vars.size());
+    assertTrue(vars.contains("a"));
+    vars = exp.getVariableNames();
+    assertEquals(1, vars.size());
+    assertTrue(vars.contains("a"));
+  }
+
   @Test
   public void testIssue77() {
     AviatorEvaluator.setOption(Options.ALWAYS_PARSE_FLOATING_POINT_NUMBER_INTO_DECIMAL, true);
