diff --git a/src/main/java/de/espend/idea/php/annotation/util/AnnotationUtil.java b/src/main/java/de/espend/idea/php/annotation/util/AnnotationUtil.java
index 3ae7dc8..d9d5bc0 100644
--- a/src/main/java/de/espend/idea/php/annotation/util/AnnotationUtil.java
+++ b/src/main/java/de/espend/idea/php/annotation/util/AnnotationUtil.java
@@ -66,6 +66,9 @@ public class AnnotationUtil {
         add("@Required");
         add("@Attributes");
         add("@Attribute");
+        add("@test");
+        add("@noninspection");
+        add("@noinspection");
     }};
 
     public static boolean isAnnotationClass(@NotNull PhpClass phpClass) {
