diff --git a/src/test/java/de/espend/idea/php/annotation/tests/inspection/AnnotationMissingUseInspectionTest.java b/src/test/java/de/espend/idea/php/annotation/tests/inspection/AnnotationMissingUseInspectionTest.java
index 2617c12..a13fc13 100644
--- a/src/test/java/de/espend/idea/php/annotation/tests/inspection/AnnotationMissingUseInspectionTest.java
+++ b/src/test/java/de/espend/idea/php/annotation/tests/inspection/AnnotationMissingUseInspectionTest.java
@@ -106,5 +106,17 @@ public class AnnotationMissingUseInspectionTest extends AnnotationLightCodeInsig
                 "}",
             AnnotationMissingUseInspection.MESSAGE
         );
+
+        assertLocalInspectionContainsNotContains("test.php", "<?php\n" +
+                "\n" +
+                "\n" +
+                "/**\n" +
+                " * @noinspe<caret>ction()\n" +
+                " */\n" +
+                "class Foo\n" +
+                "{\n" +
+                "}",
+            AnnotationMissingUseInspection.MESSAGE
+        );
     }
 }
