diff --git a/src/test/java/com/google/cloud/solutions/spannerddl/diff/DdlDiffTest.java b/src/test/java/com/google/cloud/solutions/spannerddl/diff/DdlDiffTest.java
index 5db3c5d..9e1fb18 100644
--- a/src/test/java/com/google/cloud/solutions/spannerddl/diff/DdlDiffTest.java
+++ b/src/test/java/com/google/cloud/solutions/spannerddl/diff/DdlDiffTest.java
@@ -82,7 +82,8 @@ public class DdlDiffTest {
               + "primary key (intcol ASC)");
       fail("Expected exception not thrown");
     } catch (IllegalArgumentException e) {
-      assertThat(e.getMessage()).containsMatch("anonymous FOREIGN KEY constraints");
+      assertThat(e.getMessage())
+          .containsMatch("Can not create diffs when anonymous constraints are used.");
     }
   }
 
@@ -97,7 +98,8 @@ public class DdlDiffTest {
               + "primary key (intcol ASC)");
       fail("Expected exception not thrown");
     } catch (IllegalArgumentException e) {
-      assertThat(e.getMessage()).containsMatch("anonymous CHECK constraints");
+      assertThat(e.getMessage())
+          .containsMatch("Can not create diffs when anonymous constraints are used.");
     }
   }
 
