diff --git a/src/test/java/com/github/davidmoten/rx/jdbc/DatabaseExampleTest.java b/src/test/java/com/github/davidmoten/rx/jdbc/DatabaseExampleTest.java
index 2160e2e..66bb873 100644
--- a/src/test/java/com/github/davidmoten/rx/jdbc/DatabaseExampleTest.java
+++ b/src/test/java/com/github/davidmoten/rx/jdbc/DatabaseExampleTest.java
@@ -3,7 +3,6 @@ package com.github.davidmoten.rx.jdbc;
 import static org.junit.Assert.assertEquals;
 
 import java.sql.SQLException;
-import java.util.Arrays;
 
 import org.junit.Test;
 
@@ -22,8 +21,10 @@ public class DatabaseExampleTest {
                 "create table person (name varchar(50) primary key, score int not null,dob date, registered timestamp)")
                 .count().toBlocking().single();
 
-        assertEquals(Arrays.asList(1, 1), db.update("insert into person(name,score) values(?,?)")
-                .parameters("FRED", 21, "JOE", 34).execute());
+        assertEquals(
+                2,
+                db.update("insert into person(name,score) values(?,?)")
+                        .parameters("FRED", 21, "JOE", 34).execute());
 
         // use java 8 lambdas if you have them !
         db.select("select name, score from person").autoMap(Person.class)
