diff --git a/test/org/netpreserve/jwarc/WarcParserTest.java b/test/org/netpreserve/jwarc/WarcParserTest.java
index 4216591..dec7381 100644
--- a/test/org/netpreserve/jwarc/WarcParserTest.java
+++ b/test/org/netpreserve/jwarc/WarcParserTest.java
@@ -21,6 +21,16 @@ public class WarcParserTest {
         assertEquals(Optional.of("494"), parser.headers().sole("Content-Length"));
     }
 
+    @Test
+    public void testParsingArcWithCorruptDates() {
+        WarcParser parser = parse("http://example.com/ 1.2.3.4 200012120739 text/html 42\n");
+        assertEquals(Optional.of("2000-12-12T07:39:00Z"), parser.headers().first("WARC-Date"));
+        parser = parse("http://example.com/ 1.2.3.4 2000121207394211 text/html 1942\n");
+        assertEquals(Optional.of("2000-12-12T07:39:42Z"), parser.headers().first("WARC-Date"));
+        parser = parse("http://example.com/ 1.2.3.4 99999999999999 text/html 1942\n");
+        assertEquals(Optional.empty(), parser.headers().first("WARC-Date"));
+    }
+
     private static WarcParser parse(String input) {
         WarcParser parser = new WarcParser();
         parser.parse(ByteBuffer.wrap(input.getBytes(StandardCharsets.ISO_8859_1)));
