diff --git a/src/test/java/com/alibaba/json/bvt/TestExternal3.java b/src/test/java/com/alibaba/json/bvt/TestExternal3.java
index 970b98b..e4c2dc5 100755
--- a/src/test/java/com/alibaba/json/bvt/TestExternal3.java
+++ b/src/test/java/com/alibaba/json/bvt/TestExternal3.java
@@ -16,7 +16,7 @@ import com.alibaba.fastjson.serializer.SerializerFeature;
 
 
 public class TestExternal3 extends TestCase {
-    ParserConfig confg = new ParserConfig();
+    ParserConfig confg = ParserConfig.global;
     protected void setUp() throws Exception {
         confg.addAccept("external.VO");
     }
diff --git a/src/test/java/com/alibaba/json/bvt/TestExternal4.java b/src/test/java/com/alibaba/json/bvt/TestExternal4.java
index 1489d47..702af08 100755
--- a/src/test/java/com/alibaba/json/bvt/TestExternal4.java
+++ b/src/test/java/com/alibaba/json/bvt/TestExternal4.java
@@ -17,7 +17,7 @@ import com.alibaba.fastjson.JSONObject;
 import com.alibaba.fastjson.serializer.SerializerFeature;
 
 public class TestExternal4 extends TestCase {
-    ParserConfig confg = new ParserConfig();
+    ParserConfig confg = ParserConfig.global;
     protected void setUp() throws Exception {
         confg.addAccept("external.VO2");
     }
diff --git a/src/test/java/com/alibaba/json/bvt/TestExternal5.java b/src/test/java/com/alibaba/json/bvt/TestExternal5.java
index b4b965f..529c5fe 100755
--- a/src/test/java/com/alibaba/json/bvt/TestExternal5.java
+++ b/src/test/java/com/alibaba/json/bvt/TestExternal5.java
@@ -17,7 +17,7 @@ import com.alibaba.fastjson.JSONObject;
 import com.alibaba.fastjson.serializer.SerializerFeature;
 
 public class TestExternal5 extends TestCase {
-    ParserConfig confg = new ParserConfig();
+    ParserConfig confg = ParserConfig.global;
     protected void setUp() throws Exception {
         confg.addAccept("com.alibaba.dubbo.demo");
     }
@@ -37,9 +37,11 @@ public class TestExternal5 extends TestCase {
 
         String text = JSON.toJSONString(obj, SerializerFeature.WriteClassName, SerializerFeature.WriteMapNullValue);
         System.out.println(text);
-        JSON.parseObject(text, clazz, confg);
-        String clazzName = JSON.parse(text, confg).getClass().getName();
-        Assert.assertEquals(clazz.getName(), clazzName);
+        Object object = JSON.parseObject(text, clazz, confg);
+        assertEquals("a1", clazz.getName(), object.getClass().getName());
+
+        Object object2 = JSON.parse(text, confg);
+        assertEquals("a2 " + text, clazz.getName(), object2.getClass().getName());
     }
 
     public static class ExtClassLoader extends ClassLoader {
diff --git a/src/test/java/com/alibaba/json/bvt/TestExternal6.java b/src/test/java/com/alibaba/json/bvt/TestExternal6.java
index 5d01d97..4d836c1 100755
--- a/src/test/java/com/alibaba/json/bvt/TestExternal6.java
+++ b/src/test/java/com/alibaba/json/bvt/TestExternal6.java
@@ -17,7 +17,7 @@ import com.alibaba.fastjson.JSONObject;
 import com.alibaba.fastjson.serializer.SerializerFeature;
 
 public class TestExternal6 extends TestCase {
-    ParserConfig confg = new ParserConfig();
+    ParserConfig confg = ParserConfig.global;
 
     protected void setUp() throws Exception {
         confg.addAccept("org.mule.esb.model");
