diff --git a/src/main/java/org/stellar/sdk/requests/ResponseHandler.java b/src/main/java/org/stellar/sdk/requests/ResponseHandler.java
index 49db9bc..663bff7 100644
--- a/src/main/java/org/stellar/sdk/requests/ResponseHandler.java
+++ b/src/main/java/org/stellar/sdk/requests/ResponseHandler.java
@@ -1,12 +1,12 @@
 package org.stellar.sdk.requests;
 
+import com.google.common.base.Optional;
 import com.google.gson.reflect.TypeToken;
 
 import org.stellar.sdk.responses.GsonSingleton;
 import org.stellar.sdk.responses.TypedResponse;
 
 import java.io.IOException;
-import java.util.Optional;
 
 import okhttp3.Response;
 
@@ -29,8 +29,8 @@ public class ResponseHandler<T> {
     try {
       // Too Many Requests
       if (response.code() == 429) {
-        
-        Optional<Integer> retryAfter = Optional.empty();
+
+        Optional<Integer> retryAfter = Optional.absent();
         String header = response.header("Retry-After");
         if (header != null) {
           try {
diff --git a/src/main/java/org/stellar/sdk/requests/TooManyRequestsException.java b/src/main/java/org/stellar/sdk/requests/TooManyRequestsException.java
index 921a269..c6fc72f 100644
--- a/src/main/java/org/stellar/sdk/requests/TooManyRequestsException.java
+++ b/src/main/java/org/stellar/sdk/requests/TooManyRequestsException.java
@@ -1,6 +1,7 @@
 package org.stellar.sdk.requests;
 
-import java.util.Optional;
+
+import com.google.common.base.Optional;
 
 /**
  * Exception thrown when too many requests were sent to the Horizon server.
