diff --git a/src/test/java/org/zwobble/mammoth/tests/styles/parsing/StyleMappingTokeniserTests.java b/src/test/java/org/zwobble/mammoth/tests/styles/parsing/StyleMappingTokeniserTests.java
index f3147ba..85fb6d4 100644
--- a/src/test/java/org/zwobble/mammoth/tests/styles/parsing/StyleMappingTokeniserTests.java
+++ b/src/test/java/org/zwobble/mammoth/tests/styles/parsing/StyleMappingTokeniserTests.java
@@ -51,6 +51,11 @@ public class StyleMappingTokeniserTests {
     }
 
     @Test
+    public void escapeSequencesInStringsAreTokenised() {
+        assertTokens("'Tristan\\''", isToken(TokenType.STRING, "'Tristan\\''"));
+    }
+
+    @Test
     public void unterminatedStringsAreTokenised() {
         assertTokens("'Tristan", isToken(TokenType.UNTERMINATED_STRING, "'Tristan"));
     }
