diff --git a/src/main/java/com/vonage/client/messages/MessageResponseException.java b/src/main/java/com/vonage/client/messages/MessageResponseException.java
index 1763ce1..c2f7000 100644
--- a/src/main/java/com/vonage/client/messages/MessageResponseException.java
+++ b/src/main/java/com/vonage/client/messages/MessageResponseException.java
@@ -126,6 +126,9 @@ public final class MessageResponseException extends VonageClientException {
 	 * @return An instance of this class with all known fields populated from the JSON payload, if present.
 	 */
 	public static MessageResponseException fromJson(String json) {
+		if (json == null || json.length() < 2) {
+			return new MessageResponseException();
+		}
 		try {
 			ObjectMapper mapper = new ObjectMapper();
 			return mapper.readValue(json, MessageResponseException.class);
diff --git a/src/main/java/com/vonage/client/messages/SendMessageEndpoint.java b/src/main/java/com/vonage/client/messages/SendMessageEndpoint.java
index d0f7eaf..008fe75 100644
--- a/src/main/java/com/vonage/client/messages/SendMessageEndpoint.java
+++ b/src/main/java/com/vonage/client/messages/SendMessageEndpoint.java
@@ -23,7 +23,9 @@ import org.apache.http.HttpResponse;
 import org.apache.http.client.methods.RequestBuilder;
 import org.apache.http.entity.ContentType;
 import org.apache.http.entity.StringEntity;
-import java.io.*;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
 import java.util.stream.Collectors;
 
 class SendMessageEndpoint extends AbstractMethod<MessageRequest, MessageResponse> {
@@ -62,6 +64,9 @@ class SendMessageEndpoint extends AbstractMethod<MessageRequest, MessageResponse
 				json = br.lines().collect(Collectors.joining(System.lineSeparator()));
 			}
 			MessageResponseException mrx = MessageResponseException.fromJson(json);
+			if (mrx.title == null) {
+				mrx.title = response.getStatusLine().getReasonPhrase();
+			}
 			mrx.statusCode = statusCode;
 			throw mrx;
 		}
