diff --git a/src/test/java/com/vonage/client/messages/SendMessageEndpointTest.java b/src/test/java/com/vonage/client/messages/SendMessageEndpointTest.java
index 9949bc5..c0b8173 100644
--- a/src/test/java/com/vonage/client/messages/SendMessageEndpointTest.java
+++ b/src/test/java/com/vonage/client/messages/SendMessageEndpointTest.java
@@ -21,11 +21,11 @@ import com.vonage.client.common.HttpMethod;
 import com.vonage.client.messages.sms.SmsTextRequest;
 import org.apache.http.client.methods.RequestBuilder;
 import org.apache.http.entity.ContentType;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
 import org.junit.Test;
 import java.io.ByteArrayOutputStream;
 import java.util.UUID;
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.fail;
 
 public class SendMessageEndpointTest {
 	final SendMessageEndpoint endpoint = new SendMessageEndpoint(new HttpWrapper());
@@ -60,7 +60,7 @@ public class SendMessageEndpointTest {
 	}
 
 	@Test
-	public void testParseResponse422() throws Exception {
+	public void testParseResponseFull() throws Exception {
 		final int statusCode = 422;
 		final String json = "{\n" +
 				"  \"type\": \"https://developer.nexmo.com/api-errors/messages-olympus#1120\",\n" +
@@ -79,4 +79,19 @@ public class SendMessageEndpointTest {
 			assertEquals(expected, mrx);
 		}
 	}
+
+	@Test
+	public void testParseResponseNoBody() throws Exception {
+		final int statusCode = 429;
+		try {
+			endpoint.parseResponse(TestUtils.makeJsonHttpResponse(statusCode, ""));
+			fail("Expected "+MessageResponseException.class.getName());
+		}
+		catch (MessageResponseException mrx) {
+			MessageResponseException expected = MessageResponseException.fromJson("");
+			expected.title = "OK";  // This is what TestUtils.makeJsonHttpResponse sets it to
+			expected.statusCode = statusCode;
+			assertEquals(expected, mrx);
+		}
+	}
 }
