diff --git a/src/main/java/com/searchcode/app/util/Helpers.java b/src/main/java/com/searchcode/app/util/Helpers.java
index 05211b1..b795820 100644
--- a/src/main/java/com/searchcode/app/util/Helpers.java
+++ b/src/main/java/com/searchcode/app/util/Helpers.java
@@ -309,7 +309,7 @@ public class Helpers {
     }
 
     public String replaceForIndex(String toReplace) {
-        return this.replaceNonAlphanumeric(toReplace, "_");
+        return this.replaceNonAlphanumeric(toReplace, "_").toLowerCase();
     }
 
     public String replaceNonAlphanumeric(String toReplace, String replaceWith) {
