diff --git a/Readme.html b/Readme.html
index fe9908d..f9d62a2 100644
--- a/Readme.html
+++ b/Readme.html
@@ -283,7 +283,7 @@ see javassist.Dump.
 
 <p>-version 3.27
 <ul>
-   <li>GitHub Issue #271 (PR #279), #280 (PR #281), and #282.
+   <li>GitHub Issue #271 (PR #279), #280 (PR #281), #282, and PR #294.
 </ul>
 
 <p>-version 3.26 on October 3, 2019
diff --git a/javassist.jar b/javassist.jar
index e23d414..36ce5d9 100644
Binary files a/javassist.jar and b/javassist.jar differ
diff --git a/src/main/javassist/compiler/MemberCodeGen.java b/src/main/javassist/compiler/MemberCodeGen.java
index f1ec84a..48acd13 100644
--- a/src/main/javassist/compiler/MemberCodeGen.java
+++ b/src/main/javassist/compiler/MemberCodeGen.java
@@ -365,7 +365,7 @@ public class MemberCodeGen extends CodeGen {
                 sizeExpr.accept(this);
         else
             if (sizeExpr == null) {
-                int s = init.length();
+                int s = init.size();
                 bytecode.addIconst(s);
             }
             else
@@ -414,7 +414,7 @@ public class MemberCodeGen extends CodeGen {
         }
 
         if (init != null) {
-            int s = init.length();
+            int s = init.size();
             ASTList list = init;
             for (int i = 0; i < s; i++) {
                 bytecode.addOpcode(DUP);
diff --git a/src/main/javassist/compiler/Parser.java b/src/main/javassist/compiler/Parser.java
index 5290450..ae73fb2 100644
--- a/src/main/javassist/compiler/Parser.java
+++ b/src/main/javassist/compiler/Parser.java
@@ -683,7 +683,7 @@ public final class Parser implements TokenId {
         lex.get();      // '{'
         if(lex.lookAhead() == '}'){
             lex.get();
-            return new ArrayInit(new IntConst(0,TokenId.IntConstant));
+            return new ArrayInit(null);
         }
         ASTree expr = parseExpression(tbl);
         ArrayInit init = new ArrayInit(expr);
diff --git a/src/main/javassist/compiler/ast/ArrayInit.java b/src/main/javassist/compiler/ast/ArrayInit.java
index f992f42..0b87200 100644
--- a/src/main/javassist/compiler/ast/ArrayInit.java
+++ b/src/main/javassist/compiler/ast/ArrayInit.java
@@ -25,10 +25,27 @@ public class ArrayInit extends ASTList {
     /** default serialVersionUID */
     private static final long serialVersionUID = 1L;
 
+    /**
+     * Constructs an object.
+     * @param firstElement      maybe null when the initializer is <code>{}</code> (empty).
+     */
     public ArrayInit(ASTree firstElement) {
         super(firstElement);
     }
 
+    /**
+     * Gets the number of the elements.  Don't call {@link #length()}.
+     *
+     * @return the number of the elements.
+     */
+    public int size() {
+        int s = length();
+        if (s == 1 && head() == null)
+            return 0;
+        else
+            return s;
+    }
+
     @Override
     public void accept(Visitor v) throws CompileError { v.atArrayInit(this); }
 
