diff --git a/src/test/java/com/github/davidmoten/rtree/RTreeTest.java b/src/test/java/com/github/davidmoten/rtree/RTreeTest.java
index af34fa0..e2989c9 100644
--- a/src/test/java/com/github/davidmoten/rtree/RTreeTest.java
+++ b/src/test/java/com/github/davidmoten/rtree/RTreeTest.java
@@ -116,6 +116,14 @@ public class RTreeTest {
 		assertTrue(tree.entries().contains(e2).toBlocking().single());
 	}
 
+	@Test
+	public void testDeleteItemThatIsNotPresentDoesNothing() {
+		Entry<Object> e1 = e(1);
+		Entry<Object> e2 = e(2);
+		RTree<Object> tree = RTree.builder().build().add(e1);
+		assertEquals(tree, tree.delete(e2));
+	}
+
 	private static Entry<Object> e(int n) {
 		return new Entry<Object>(new Object(), r(n));
 	}
