diff --git a/src/main/java/nodebox/node/Port.java b/src/main/java/nodebox/node/Port.java
index d745a66..ccfce39 100644
--- a/src/main/java/nodebox/node/Port.java
+++ b/src/main/java/nodebox/node/Port.java
@@ -799,7 +799,8 @@ public final class Port {
                 && Objects.equal(type, other.type)
                 && Objects.equal(label, other.label)
                 && Objects.equal(value, other.value)
-                && Objects.equal(description, other.description);
+                && Objects.equal(description, other.description)
+                && Objects.equal(range, other.range);
     }
 
     @Override
