diff --git a/src/main/java/com/projecty/projectyweb/ProjectyWebApplication.java b/src/main/java/com/projecty/projectyweb/ProjectyWebApplication.java
index 217433e..a7772ad 100644
--- a/src/main/java/com/projecty/projectyweb/ProjectyWebApplication.java
+++ b/src/main/java/com/projecty/projectyweb/ProjectyWebApplication.java
@@ -2,8 +2,10 @@ package com.projecty.projectyweb;
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.data.jpa.repository.config.EnableJpaAuditing;
 
 @SpringBootApplication
+@EnableJpaAuditing
 public class ProjectyWebApplication {
 
     public static void main(String[] args) {
diff --git a/src/main/java/com/projecty/projectyweb/user/User.java b/src/main/java/com/projecty/projectyweb/user/User.java
index a256530..90d070c 100644
--- a/src/main/java/com/projecty/projectyweb/user/User.java
+++ b/src/main/java/com/projecty/projectyweb/user/User.java
@@ -6,6 +6,7 @@ import com.projecty.projectyweb.project.role.ProjectRole;
 import com.projecty.projectyweb.task.Task;
 import com.projecty.projectyweb.team.role.TeamRole;
 import org.springframework.data.annotation.CreatedDate;
+import org.springframework.data.jpa.domain.support.AuditingEntityListener;
 
 import javax.persistence.*;
 import java.io.Serializable;
@@ -13,6 +14,7 @@ import java.time.LocalDateTime;
 import java.util.List;
 @Entity
 @JsonSerialize(using = UserSerializer.class)
+@EntityListeners(AuditingEntityListener.class)
 public class User implements Serializable {
     @Id
     @GeneratedValue
