diff --git a/src/main/java/com/projecty/projectyweb/ProjectyWebApplication.java b/src/main/java/com/projecty/projectyweb/ProjectyWebApplication.java
index a7772ad..f7e4220 100644
--- a/src/main/java/com/projecty/projectyweb/ProjectyWebApplication.java
+++ b/src/main/java/com/projecty/projectyweb/ProjectyWebApplication.java
@@ -2,10 +2,9 @@ package com.projecty.projectyweb;
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
-import org.springframework.data.jpa.repository.config.EnableJpaAuditing;
 
 @SpringBootApplication
-@EnableJpaAuditing
+//@EnableJpaAuditing
 public class ProjectyWebApplication {
 
     public static void main(String[] args) {
diff --git a/src/main/java/com/projecty/projectyweb/user/User.java b/src/main/java/com/projecty/projectyweb/user/User.java
index 90d070c..8db6565 100644
--- a/src/main/java/com/projecty/projectyweb/user/User.java
+++ b/src/main/java/com/projecty/projectyweb/user/User.java
@@ -6,7 +6,6 @@ import com.projecty.projectyweb.project.role.ProjectRole;
 import com.projecty.projectyweb.task.Task;
 import com.projecty.projectyweb.team.role.TeamRole;
 import org.springframework.data.annotation.CreatedDate;
-import org.springframework.data.jpa.domain.support.AuditingEntityListener;
 
 import javax.persistence.*;
 import java.io.Serializable;
@@ -14,7 +13,7 @@ import java.time.LocalDateTime;
 import java.util.List;
 @Entity
 @JsonSerialize(using = UserSerializer.class)
-@EntityListeners(AuditingEntityListener.class)
+//@EntityListeners(AuditingEntityListener.class)
 public class User implements Serializable {
     @Id
     @GeneratedValue
