diff --git a/src/main/java/io/r2dbc/proxy/listener/LifeCycleExecutionListener.java b/src/main/java/io/r2dbc/proxy/listener/LifeCycleExecutionListener.java
index 97185ea..3d86d5c 100644
--- a/src/main/java/io/r2dbc/proxy/listener/LifeCycleExecutionListener.java
+++ b/src/main/java/io/r2dbc/proxy/listener/LifeCycleExecutionListener.java
@@ -183,6 +183,12 @@ public final class LifeCycleExecutionListener implements ProxyExecutionListener 
                 } else {
                     this.delegate.afterExecuteOnStatement(executionInfo);
                 }
+            } else if ("returnGeneratedValues".equals(methodName)) {
+                if (isBefore) {
+                    this.delegate.beforeReturnGeneratedValuesOnStatement(executionInfo);
+                } else {
+                    this.delegate.afterReturnGeneratedValuesOnStatement(executionInfo);
+                }
             }
         } else if (Result.class.equals(methodDeclaringClass)) {
             if ("getRowsUpdated".equals(methodName)) {
diff --git a/src/main/java/io/r2dbc/proxy/listener/LifeCycleListener.java b/src/main/java/io/r2dbc/proxy/listener/LifeCycleListener.java
index 3427474..60bc67e 100644
--- a/src/main/java/io/r2dbc/proxy/listener/LifeCycleListener.java
+++ b/src/main/java/io/r2dbc/proxy/listener/LifeCycleListener.java
@@ -339,6 +339,22 @@ public interface LifeCycleListener {
     default void afterExecuteOnStatement(MethodExecutionInfo methodExecutionInfo) {
     }
 
+    /**
+     * Callback that is invoked <em>before</em> {@link Statement#returnGeneratedValues(String...)} is called.
+     *
+     * @param methodExecutionInfo the current method execution info; never {@code null}.
+     */
+    default void beforeReturnGeneratedValuesOnStatement(MethodExecutionInfo methodExecutionInfo) {
+    }
+
+    /**
+     * Callback that is invoked <em>after</em> {@link Statement#returnGeneratedValues(String...)} is called.
+     *
+     * @param methodExecutionInfo the current method execution info; never {@code null}.
+     */
+    default void afterReturnGeneratedValuesOnStatement(MethodExecutionInfo methodExecutionInfo) {
+    }
+
     //
     // For Result
     //
