diff --git a/data_for_test.go b/data_for_test.go
index 506ce9d..0fb14f6 100644
--- a/data_for_test.go
+++ b/data_for_test.go
@@ -327,7 +327,7 @@ var suites = []FixtureSuite{
 		{
 			Name: "otel-cli span background (recording)",
 			Config: FixtureConfig{
-				CliArgs:       []string{"span", "background", "--timeout", "1s", "--sockdir", "."},
+				CliArgs:       []string{"span", "background", "--timeout", "1s", "--sockdir", ".", "--attrs", "abc=def"},
 				Env:           map[string]string{"OTEL_EXPORTER_OTLP_ENDPOINT": "{{endpoint}}"},
 				TestTimeoutMs: 2000,
 				Background:    true,
@@ -336,8 +336,9 @@ var suites = []FixtureSuite{
 			Expect: Results{
 				Config: otelcli.DefaultConfig(),
 				SpanData: map[string]string{
-					"span_id":  "*",
-					"trace_id": "*",
+					"span_id":    "*",
+					"trace_id":   "*",
+					"attributes": `abc=def`, // weird format because of limitation in OTLP server
 				},
 				Spans:  1,
 				Events: 1,
