diff --git a/src/main/java/org/traccar/protocol/PuiProtocolDecoder.java b/src/main/java/org/traccar/protocol/PuiProtocolDecoder.java
index a80af65..f10ff3f 100644
--- a/src/main/java/org/traccar/protocol/PuiProtocolDecoder.java
+++ b/src/main/java/org/traccar/protocol/PuiProtocolDecoder.java
@@ -51,7 +51,7 @@ public class PuiProtocolDecoder extends BaseMqttProtocolDecoder {
 
                 position.setValid(true);
 
-                DateFormat dateFormat = new SimpleDateFormat("yyyyMMdd'T'HHmmss.SSS'Z'");
+                DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
                 position.setTime(dateFormat.parse(json.getString("ts")));
 
                 JsonObject location = json.getJsonObject("location");
