diff --git a/src/main/java/net/masterthought/cucumber/generators/FeatureReportPage.java b/src/main/java/net/masterthought/cucumber/generators/FeatureReportPage.java
index d445a47..208afe9 100644
--- a/src/main/java/net/masterthought/cucumber/generators/FeatureReportPage.java
+++ b/src/main/java/net/masterthought/cucumber/generators/FeatureReportPage.java
@@ -3,6 +3,7 @@ package net.masterthought.cucumber.generators;
 import net.masterthought.cucumber.Configuration;
 import net.masterthought.cucumber.ReportResult;
 import net.masterthought.cucumber.json.Feature;
+import net.masterthought.cucumber.presentation.PresentationMode;
 
 public class FeatureReportPage extends AbstractPage {
 
@@ -21,6 +22,7 @@ public class FeatureReportPage extends AbstractPage {
     @Override
     public void prepareReport() {
         context.put("feature", feature);
+        context.put("parallel_testing", configuration.containsPresentationMode(PresentationMode.PARALLEL_TESTING));
     }
 
 }
