diff --git a/src/test/java/net/masterthought/cucumber/generators/FeatureReportPageTest.java b/src/test/java/net/masterthought/cucumber/generators/FeatureReportPageTest.java
index 6c29c6b..5a06913 100644
--- a/src/test/java/net/masterthought/cucumber/generators/FeatureReportPageTest.java
+++ b/src/test/java/net/masterthought/cucumber/generators/FeatureReportPageTest.java
@@ -47,7 +47,7 @@ public class FeatureReportPageTest extends PageTest {
 
         // then
         VelocityContext context = page.context;
-        assertThat(context.getKeys()).hasSize(14);
+        assertThat(context.getKeys()).hasSize(15);
         assertThat(context.get("feature")).isEqualTo(feature);
     }
 
diff --git a/src/test/java/net/masterthought/cucumber/generators/integrations/FeatureReportPageIntegrationTest.java b/src/test/java/net/masterthought/cucumber/generators/integrations/FeatureReportPageIntegrationTest.java
index 91d5713..e3fba70 100644
--- a/src/test/java/net/masterthought/cucumber/generators/integrations/FeatureReportPageIntegrationTest.java
+++ b/src/test/java/net/masterthought/cucumber/generators/integrations/FeatureReportPageIntegrationTest.java
@@ -1,5 +1,9 @@
 package net.masterthought.cucumber.generators.integrations;
 
+import java.io.File;
+import java.util.Arrays;
+import java.util.Collections;
+
 import net.masterthought.cucumber.generators.FeatureReportPage;
 import net.masterthought.cucumber.generators.integrations.helpers.BriefAssertion;
 import net.masterthought.cucumber.generators.integrations.helpers.DocumentAssertion;
@@ -22,13 +26,10 @@ import net.masterthought.cucumber.json.Output;
 import net.masterthought.cucumber.json.Result;
 import net.masterthought.cucumber.json.Row;
 import net.masterthought.cucumber.json.Step;
+import net.masterthought.cucumber.presentation.PresentationMode;
 import org.apache.commons.lang.StringUtils;
 import org.junit.Test;
 
-import java.io.File;
-import java.util.Arrays;
-import java.util.Collections;
-
 import static org.assertj.core.api.Assertions.assertThat;
 
 /**
@@ -112,6 +113,26 @@ public class FeatureReportPageIntegrationTest extends PageTest {
     }
 
     @Test
+    public void generatePage_OnParallelTesting_generatesQualifierColumn() {
+
+        // given
+        setUpWithJson(SAMPLE_JSON);
+        final Feature feature = features.get(0);
+        configuration.addPresentationModes(PresentationMode.PARALLEL_TESTING);
+        page = new FeatureReportPage(reportResult, configuration, feature);
+
+        // when
+        page.generatePage();
+
+        // then
+        DocumentAssertion document = documentFrom(page.getWebPage());
+        TableRowAssertion bodyRow = document.getReport().getTableStats().getBodyRow();
+
+        bodyRow.hasExactValues(feature.getName(), "sample", "10", "0", "0", "0", "0", "10", "1", "0", "1", "1:39.263", "Passed");
+        bodyRow.hasExactCSSClasses("tagname", "", "passed", "", "", "", "", "total", "passed", "", "total", "duration", "passed");
+    }
+
+    @Test
     public void generatePage_generatesFeatureDetails() {
 
         // given
