diff --git a/src/test/java/com/thealgorithms/maths/FactorialTest.java b/src/test/java/com/thealgorithms/maths/FactorialTest.java
index 752cd7a..b22ad53 100644
--- a/src/test/java/com/thealgorithms/maths/FactorialTest.java
+++ b/src/test/java/com/thealgorithms/maths/FactorialTest.java
@@ -5,9 +5,19 @@ import static org.junit.jupiter.api.Assertions.*;
 import org.junit.jupiter.api.Test;
 
 public class FactorialTest {
+    private static final String EXCEPTION_MESSAGE = "Input number cannot be negative";
 
     @Test
-    public void test() {
+    public void testWhenInvalidInoutProvidedShouldThrowException() {
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> Factorial.factorial(-1));
+        assertEquals(exception.getMessage(), EXCEPTION_MESSAGE);
+    }
+
+    @Test
+    public void testCorrectFactorialCalculation() {
+        assertEquals(1, Factorial.factorial(0));
+        assertEquals(1, Factorial.factorial(1));
         assertEquals(120, Factorial.factorial(5));
+        assertEquals(3628800, Factorial.factorial(10));
     }
 }
