diff --git a/src/test/java/com/googlecode/aviator/test/function/FunctionTest.java b/src/test/java/com/googlecode/aviator/test/function/FunctionTest.java
index 097f01b..2f16f39 100644
--- a/src/test/java/com/googlecode/aviator/test/function/FunctionTest.java
+++ b/src/test/java/com/googlecode/aviator/test/function/FunctionTest.java
@@ -826,6 +826,17 @@ public class FunctionTest {
     assertEquals("a.b", vars.get(0));
   }
 
+  @Test
+  public void testIssue431() {
+    Expression expr = this.instance.compile("{\n" + "    let itemv1 = seq.get(info.value,index);\n"
+        + "    print( itemv1.name )\n" + "}");
+
+    List<String> vars = expr.getVariableNames();
+    assertEquals(2, vars.size());
+    assertEquals("info", vars.get(0));
+    assertEquals("index", vars.get(1));
+  }
+
 
   @Test
   public void testGetVariableNames() {
