diff --git a/src/org/joni/ByteCodeMachine.java b/src/org/joni/ByteCodeMachine.java
index 37ecbe1..7c4aa8e 100644
--- a/src/org/joni/ByteCodeMachine.java
+++ b/src/org/joni/ByteCodeMachine.java
@@ -1320,6 +1320,7 @@ class ByteCodeMachine extends StackMachine {
     private void opMemoryStart() {
         int mem = code[ip++];
         repeatStk[memStartStk + mem] = s;
+        repeatStk[memEndStk + mem] = -1;
     }
 
     private void opMemoryEndPush() {
