diff --git a/CHANGELOG.md b/CHANGELOG.md
index 7763b85..5e92219 100644
--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,7 @@
 
 - [changed] Removed org.json dependency and replaced with com.google.code.gson.
 - [changed] Upgraded Mockito dependency, and fixed the build on Java 11.
+- [fixed] UploadAccountResponse now has the correct error message/reason.
 
 # v6.7.0
 
diff --git a/src/main/java/com/google/firebase/auth/internal/UploadAccountResponse.java b/src/main/java/com/google/firebase/auth/internal/UploadAccountResponse.java
index abf2853..544617c 100644
--- a/src/main/java/com/google/firebase/auth/internal/UploadAccountResponse.java
+++ b/src/main/java/com/google/firebase/auth/internal/UploadAccountResponse.java
@@ -35,7 +35,7 @@ public class UploadAccountResponse {
     @Key("index")
     private int index;
 
-    @Key("message")
+    @Key("errorMessage")
     private String message;
 
     public int getIndex() {
