diff --git a/src/test/java/no/bekk/bekkopen/person/FodselsnummerCalculatorTest.java b/src/test/java/no/bekk/bekkopen/person/FodselsnummerCalculatorTest.java
index 8820d68..c73cc8c 100755
--- a/src/test/java/no/bekk/bekkopen/person/FodselsnummerCalculatorTest.java
+++ b/src/test/java/no/bekk/bekkopen/person/FodselsnummerCalculatorTest.java
@@ -39,7 +39,7 @@ public class FodselsnummerCalculatorTest {
 	@Test
 	public void getValidFodselsnummerForDate() {
 		List<Fodselsnummer> validOptions = FodselsnummerCalculator.getManyFodselsnummerForDate(date);
-		assertTrue("Forventet 38 fødselsnumre, men fikk " + validOptions.size(), validOptions.size() == 38);
+		assertTrue("Forventet 413 fødselsnumre, men fikk " + validOptions.size(), validOptions.size() == 413);
 	}
 
    @Test
diff --git a/src/test/java/no/bekk/bekkopen/person/FodselsnummerValidatorTest.java b/src/test/java/no/bekk/bekkopen/person/FodselsnummerValidatorTest.java
index fdfe487..52f3556 100644
--- a/src/test/java/no/bekk/bekkopen/person/FodselsnummerValidatorTest.java
+++ b/src/test/java/no/bekk/bekkopen/person/FodselsnummerValidatorTest.java
@@ -1,5 +1,6 @@
 package no.bekk.bekkopen.person;
 
+import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
 import no.bekk.bekkopen.NoCommonsTestCase;
@@ -117,4 +118,12 @@ public class FodselsnummerValidatorTest extends NoCommonsTestCase {
 	public void testGetDNumber() {
 		FodselsnummerValidator.getFodselsnummer("47086303651");
 	}
+
+	@Test
+	public void testCalculateSecondChecksumDigit() {
+		Fodselsnummer testcase = new Fodselsnummer("1234567891");
+		int correctChecksum = 1;
+		int calculatedChecksum = FodselsnummerValidator.calculateSecondChecksumDigit(testcase);
+		assertEquals(calculatedChecksum, correctChecksum);
+	}
 }
