diff --git a/src/test/java/com/jitterted/mobreg/domain/EnsembleMembersTest.java b/src/test/java/com/jitterted/mobreg/domain/EnsembleMembersTest.java
index 569a35a..cd27e0c 100644
--- a/src/test/java/com/jitterted/mobreg/domain/EnsembleMembersTest.java
+++ b/src/test/java/com/jitterted/mobreg/domain/EnsembleMembersTest.java
@@ -60,6 +60,20 @@ public class EnsembleMembersTest {
     }
 
     @Test
+    void spectatorWhenAcceptRemovesFromSpectators() {
+        Ensemble ensemble = EnsembleFactory.withStartTimeNow();
+        MemberId memberId = MemberId.of(123);
+        ensemble.joinAsSpectator(memberId);
+
+        ensemble.acceptedBy(memberId);
+
+        assertThat(ensemble.spectators())
+                .isEmpty();
+        assertThat(ensemble.acceptedMembers())
+                .containsExactly(memberId);
+    }
+
+    @Test
     public void acceptMemberByIdWithEnsembleRemembersTheMember() throws Exception {
         Ensemble ensemble = EnsembleFactory.withStartTimeNow();
         MemberId memberId = MemberId.of(123);
