diff --git a/src/test/java/com/thoughtworks/qdox/model/JavaFieldTest.java b/src/test/java/com/thoughtworks/qdox/model/JavaFieldTest.java
index 4c33875..2c3eaff 100644
--- a/src/test/java/com/thoughtworks/qdox/model/JavaFieldTest.java
+++ b/src/test/java/com/thoughtworks/qdox/model/JavaFieldTest.java
@@ -220,4 +220,16 @@ public abstract class JavaFieldTest<F extends JavaField> extends TestCase {
         setModifiers( fld, Arrays.asList( new String[] { "volatile" } ) );
         assertTrue( fld.isVolatile() );
     }
+    
+    public void testEquals()
+    {
+        JavaClass type = mock(JavaClass.class);
+        JavaClass declaringClass = mock(JavaClass.class);
+        F fld1 = newJavaField( type, "field");
+        F fld2 = newJavaField( type, "field");
+        setDeclaringClass( fld1, declaringClass );
+        setDeclaringClass( fld2, declaringClass );
+        
+        assertEquals( fld1, fld2 );
+    }
 }
\ No newline at end of file
