diff --git a/src/main/java/io/github/mngsk/devicedetector/Detection.java b/src/main/java/io/github/mngsk/devicedetector/Detection.java
index 8e1cebb..f67b49a 100644
--- a/src/main/java/io/github/mngsk/devicedetector/Detection.java
+++ b/src/main/java/io/github/mngsk/devicedetector/Detection.java
@@ -28,7 +28,7 @@ public class Detection {
 	private static Pattern chromePattern = Pattern.compile(
 			"(?:^|[^A-Z_-])(?:Chrome/[\\.0-9]*)", Pattern.CASE_INSENSITIVE);
 	private static Pattern chromeSmartphonePattern = Pattern.compile(
-			"(?:^|[^A-Z_-])(?:Chrome/[\\.0-9]* Mobile)",
+			"(?:^|[^A-Z_-])(?:Chrome/[\\.0-9]* (?:Mobile|eliboM))",
 			Pattern.CASE_INSENSITIVE);
 	private static Pattern chromeTabletPattern = Pattern.compile(
 			"(?:^|[^A-Z_-])(?:Chrome/[\\.0-9]* (?!Mobile))",
