diff --git a/src/main/java/net/revelc/code/formatter/jsoup/JsoupBasedFormatter.java b/src/main/java/net/revelc/code/formatter/jsoup/JsoupBasedFormatter.java
index 9a1edb3..b638a73 100644
--- a/src/main/java/net/revelc/code/formatter/jsoup/JsoupBasedFormatter.java
+++ b/src/main/java/net/revelc/code/formatter/jsoup/JsoupBasedFormatter.java
@@ -102,6 +102,12 @@ public abstract class JsoupBasedFormatter extends AbstractCacheableFormatter imp
             formattedCode = String.join(ending.getChars(), newLines);
         }
 
+        // XXX: Adding new line at end of file until jsoup fixes bug.
+        String[] lines = formattedCode.split(ending.getChars());
+        if (!lines[lines.length - 1].equals(ending.getChars())) {
+            formattedCode = formattedCode + ending.getChars();
+        }
+
         if (code.equals(formattedCode)) {
             return null;
         }
