diff --git a/src/test/java/no/bekk/bekkopen/banking/KidnummerValidatorTest.java b/src/test/java/no/bekk/bekkopen/banking/KidnummerValidatorTest.java
index 5c40644..1caad93 100644
--- a/src/test/java/no/bekk/bekkopen/banking/KidnummerValidatorTest.java
+++ b/src/test/java/no/bekk/bekkopen/banking/KidnummerValidatorTest.java
@@ -1,13 +1,12 @@
 package no.bekk.bekkopen.banking;
 
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertTrue;
-import static org.junit.Assert.fail;
 import no.bekk.bekkopen.NoCommonsTestCase;
 import no.bekk.bekkopen.common.StringNumberValidator;
 
 import org.junit.Test;
 
+import static org.junit.Assert.*;
+
 public class KidnummerValidatorTest extends NoCommonsTestCase {
 
 	private static final String KIDNUMMER_VALID_MOD10 = "2345676";
@@ -66,6 +65,13 @@ public class KidnummerValidatorTest extends NoCommonsTestCase {
 		}
 	}
 
+    @Test
+    public void testValidKidnummerMod10ButUnableToCalculateMod11() {
+        boolean result = KidnummerValidator.isValid("01290865");
+
+        assertEquals(true, result);
+    }
+
 	@Test
 	public void testIsValidMod10() {
 		assertTrue(KidnummerValidator.isValid(KIDNUMMER_VALID_MOD10));
