diff --git a/src/test/java/com/thoughtworks/qdox/JavaProjectBuilderTest.java b/src/test/java/com/thoughtworks/qdox/JavaProjectBuilderTest.java
index 7533629..fd9cc83 100644
--- a/src/test/java/com/thoughtworks/qdox/JavaProjectBuilderTest.java
+++ b/src/test/java/com/thoughtworks/qdox/JavaProjectBuilderTest.java
@@ -1,6 +1,5 @@
 package com.thoughtworks.qdox;
 
-import static org.junit.Assert.assertTrue;
 import static org.mockito.Matchers.any;
 import static org.mockito.Matchers.same;
 import static org.mockito.Mockito.mock;
@@ -1670,6 +1669,24 @@ public class JavaProjectBuilderTest extends TestCase
         assertNotNull(clss);
     }
     
+    // github #75
+    public void testAnnotationWithConstant() {  
+        String constantSource = "package com.xenoamess;\n"
+                        + "public interface Constants {\n"
+                        + "  public String STATIC_STRING_A=\"SOME_VALUE\";" + "}";
+        builder.addSource( new StringReader( constantSource ) );
+
+        String source = "import static com.xenoamess.Constants.STATIC_STRING_A;\n"
+                        + "\n" + "public class ClassA{\n"
+                        + "    @AnnotationA(annotationFieldA = STATIC_STRING_A)\n"
+                        + "    public void functionA(){\n" + "    }\n"
+                        + "}";
+        JavaMethod method =
+            builder.addSource( new StringReader( source ) ).getClassByName( "ClassA" ).getMethods().get( 0 );
+        assertEquals( "@AnnotationA(annotationFieldA=com.xenoamess.Constants.STATIC_STRING_A)\n",
+                      method.getAnnotations().get( 0 ).getCodeBlock() );
+    }
+    
     // Github #76
     public void testAtInComment() throws Exception 
     {
diff --git a/src/test/java/com/thoughtworks/qdox/builder/impl/EvaluatingVisitorTest.java b/src/test/java/com/thoughtworks/qdox/builder/impl/EvaluatingVisitorTest.java
index 29dd16e..e0c7737 100644
--- a/src/test/java/com/thoughtworks/qdox/builder/impl/EvaluatingVisitorTest.java
+++ b/src/test/java/com/thoughtworks/qdox/builder/impl/EvaluatingVisitorTest.java
@@ -493,11 +493,10 @@ public class EvaluatingVisitorTest
         }
         
         JavaField nonStaticNonFinalfield = mock( JavaField.class );
-        
         JavaClass declaringClass = mock( JavaClass.class );
         when( declaringClass.getFieldByName( "fieldname" ) ).thenReturn( nonStaticNonFinalfield );
+        when( nonStaticNonFinalfield.getDeclaringClass() ).thenReturn( declaringClass );
 
-        JavaClass annotatedClass = mock( JavaClass.class );
         ref.setDeclaringClass( declaringClass );
         try 
         {
