diff --git a/src/test/java/org/stellar/sdk/KeyPairTest.java b/src/test/java/org/stellar/sdk/KeyPairTest.java
index 680484c..eb2b655 100644
--- a/src/test/java/org/stellar/sdk/KeyPairTest.java
+++ b/src/test/java/org/stellar/sdk/KeyPairTest.java
@@ -120,4 +120,25 @@ public class KeyPairTest {
     // the hint could only be derived off of 3 bytes from payload
     Assert.assertArrayEquals(sig.getHint().getSignatureHint(), new byte[]{(byte)(255), 64, 7, 55});
   }
+
+  @Test
+  public void testPublicEqual() {
+    KeyPair keypair = KeyPair.fromAccountId("GDEAOZWTVHQZGGJY6KG4NAGJQ6DXATXAJO3AMW7C4IXLKMPWWB4FDNFZ");
+    KeyPair keypairCopy = KeyPair.fromAccountId("GDEAOZWTVHQZGGJY6KG4NAGJQ6DXATXAJO3AMW7C4IXLKMPWWB4FDNFZ");
+    Assert.assertEquals(keypairCopy, keypair);
+  }
+
+  @Test
+  public void testPublicPrivateNotEquals() {
+    KeyPair keypair = KeyPair.random();
+    KeyPair keypairPublicCopy = KeyPair.fromAccountId(keypair.getAccountId());
+    Assert.assertNotEquals(keypairPublicCopy, keypair);
+  }
+
+  @Test
+  public void testPrivateEquals() {
+    KeyPair keyPair = KeyPair.random();
+    KeyPair keypairCopy = KeyPair.fromSecretSeed(keyPair.getSecretSeed());
+    Assert.assertEquals(keyPair, keypairCopy);
+  }
 }
