diff --git a/src/main/java/org/mvel2/util/ParseTools.java b/src/main/java/org/mvel2/util/ParseTools.java
index e055ead..1bd2ee2 100644
--- a/src/main/java/org/mvel2/util/ParseTools.java
+++ b/src/main/java/org/mvel2/util/ParseTools.java
@@ -1042,6 +1042,7 @@ public class ParseTools {
 
     typeCodes.put(BigDecimal.class, DataTypes.BIG_DECIMAL);
     typeCodes.put(BigInteger.class, DataTypes.BIG_INTEGER);
+    typeCodes.put(InternalNumber.class, DataTypes.BIG_DECIMAL);
 
     typeCodes.put(int.class, DataTypes.INTEGER);
     typeCodes.put(double.class, DataTypes.DOUBLE);
