diff --git a/src/test/java/com/vonage/client/insight/BasicInsightEndpointTest.java b/src/test/java/com/vonage/client/insight/BasicInsightEndpointTest.java
index bfce606..f31e55c 100644
--- a/src/test/java/com/vonage/client/insight/BasicInsightEndpointTest.java
+++ b/src/test/java/com/vonage/client/insight/BasicInsightEndpointTest.java
@@ -52,7 +52,6 @@ public class BasicInsightEndpointTest {
         Map<String, String> params = TestUtils.makeParameterMap(builder.getParameters());
         assertEquals("1234", params.get("number"));
         assertNull(params.get("country"));
-        assertEquals(ContentType.APPLICATION_JSON.getMimeType(), builder.getFirstHeader("Content-Type").getValue());
         assertEquals(ContentType.APPLICATION_JSON.getMimeType(), builder.getFirstHeader("Accept").getValue());
     }
 
diff --git a/src/test/java/com/vonage/client/insight/StandardInsightEndpointTest.java b/src/test/java/com/vonage/client/insight/StandardInsightEndpointTest.java
index 2518ef2..442bded 100644
--- a/src/test/java/com/vonage/client/insight/StandardInsightEndpointTest.java
+++ b/src/test/java/com/vonage/client/insight/StandardInsightEndpointTest.java
@@ -50,7 +50,6 @@ public class StandardInsightEndpointTest {
         assertEquals("POST", builder.getMethod());
         assertEquals("https://api.nexmo.com/ni/standard/json", builder.build().getURI().toString());
         assertEquals(ContentType.APPLICATION_JSON.getMimeType(), builder.getFirstHeader("Accept").getValue());
-        assertEquals(ContentType.APPLICATION_JSON.getMimeType(), builder.getFirstHeader("Content-Type").getValue());
         Map<String, String> params = TestUtils.makeParameterMap(builder.getParameters());
         assertEquals("1234", params.get("number"));
         assertNull(params.get("country"));
diff --git a/src/test/java/com/vonage/client/numbers/UpdateNumberEndpointTest.java b/src/test/java/com/vonage/client/numbers/UpdateNumberEndpointTest.java
index 1f85be7..4a17d35 100644
--- a/src/test/java/com/vonage/client/numbers/UpdateNumberEndpointTest.java
+++ b/src/test/java/com/vonage/client/numbers/UpdateNumberEndpointTest.java
@@ -21,6 +21,7 @@ import com.vonage.client.TestUtils;
 import com.vonage.client.auth.TokenAuthMethod;
 import org.apache.http.HttpResponse;
 import org.apache.http.client.methods.RequestBuilder;
+import org.apache.http.entity.ContentType;
 import org.junit.Before;
 import org.junit.Test;
 
@@ -48,6 +49,7 @@ public class UpdateNumberEndpointTest {
         assertEquals("POST", builder.getMethod());
         assertEquals("https://rest.nexmo.com/number/update", builder.build().getURI().toString());
 
+        assertEquals(ContentType.APPLICATION_JSON.getMimeType(), builder.getFirstHeader("Accept").getValue());
         Map<String, String> params = TestUtils.makeParameterMap(builder.getParameters());
         assertEquals(2, params.size());
         assertEquals("447700900013", params.get("msisdn"));
diff --git a/src/test/java/com/vonage/client/voice/StopTalkMethodTest.java b/src/test/java/com/vonage/client/voice/StopTalkMethodTest.java
index 17a29d6..9bfc66e 100644
--- a/src/test/java/com/vonage/client/voice/StopTalkMethodTest.java
+++ b/src/test/java/com/vonage/client/voice/StopTalkMethodTest.java
@@ -52,7 +52,6 @@ public class StopTalkMethodTest {
         RequestBuilder request = method.makeRequest("63f61863-4a51-4f6b-86e1-46edebcf9356");
 
         assertEquals("DELETE", request.getMethod());
-        assertEquals(ContentType.APPLICATION_JSON.getMimeType(), request.getFirstHeader("Content-Type").getValue());
         assertEquals(ContentType.APPLICATION_JSON.getMimeType(), request.getFirstHeader("Accept").getValue());
 
         assertTrue(request.getUri().toString().contains("63f61863-4a51-4f6b-86e1-46edebcf9356"));
@@ -86,6 +85,5 @@ public class StopTalkMethodTest {
         RequestBuilder builder = method.makeRequest("63f61863-4a51-4f6b-86e1-46edebcf9356");
 
         assertEquals(expectedUri, builder.getUri().toString());
-
     }
 }
\ No newline at end of file
