diff --git a/src/main/java/com/googlecode/aviator/lexer/ExpressionLexer.java b/src/main/java/com/googlecode/aviator/lexer/ExpressionLexer.java
index c7ea47d..4b774ad 100644
--- a/src/main/java/com/googlecode/aviator/lexer/ExpressionLexer.java
+++ b/src/main/java/com/googlecode/aviator/lexer/ExpressionLexer.java
@@ -221,6 +221,7 @@ public class ExpressionLexer {
             int n = digit;
             nextChar();
             while (Character.isDigit(this.peek)) {
+              sb.append(this.peek);
               n = 10 * n + Character.digit(this.peek, 10);
               nextChar();
             }
@@ -250,6 +251,8 @@ public class ExpressionLexer {
 
       } while (Character.isDigit(this.peek) || this.peek == '.' || this.peek == 'E'
           || this.peek == 'e' || this.peek == 'M' || this.peek == 'N');
+
+
       Number value;
       if (isBigDecimal) {
         value = new BigDecimal(getBigNumberLexeme(sb), this.mathContext);
