diff --git a/src/main/java/de/espend/idea/php/annotation/doctrine/inspection/DoctrineTypeDeprecatedInspection.java b/src/main/java/de/espend/idea/php/annotation/doctrine/inspection/DoctrineTypeDeprecatedInspection.java
index 2698440..e42fa82 100644
--- a/src/main/java/de/espend/idea/php/annotation/doctrine/inspection/DoctrineTypeDeprecatedInspection.java
+++ b/src/main/java/de/espend/idea/php/annotation/doctrine/inspection/DoctrineTypeDeprecatedInspection.java
@@ -82,7 +82,7 @@ public class DoctrineTypeDeprecatedInspection extends LocalInspectionTool {
 
                             holder.registerProblem(
                                 element,
-                                deprecationMessage != null ? deprecationMessage : String.format("Field '%s' is deprecated", text),
+                                "[Annotations] " + (deprecationMessage != null ? deprecationMessage : String.format("Field '%s' is deprecated", text)),
                                 ProblemHighlightType.LIKE_DEPRECATED
                             );
 
diff --git a/src/main/java/de/espend/idea/php/annotation/doctrine/inspection/RepositoryClassInspection.java b/src/main/java/de/espend/idea/php/annotation/doctrine/inspection/RepositoryClassInspection.java
index 9009e29..9c823ae 100644
--- a/src/main/java/de/espend/idea/php/annotation/doctrine/inspection/RepositoryClassInspection.java
+++ b/src/main/java/de/espend/idea/php/annotation/doctrine/inspection/RepositoryClassInspection.java
@@ -24,8 +24,7 @@ import java.util.Map;
  * @author Daniel Espendiller <daniel@espendiller.net>
  */
 public class RepositoryClassInspection extends LocalInspectionTool {
-
-    public static final String MESSAGE = "Missing repository class";
+    public static final String MESSAGE = "[Annotations] Missing repository class";
 
     @NotNull
     @Override
diff --git a/src/main/java/de/espend/idea/php/annotation/inspection/AnnotationDeprecatedInspection.java b/src/main/java/de/espend/idea/php/annotation/inspection/AnnotationDeprecatedInspection.java
index 92679a9..44130c6 100644
--- a/src/main/java/de/espend/idea/php/annotation/inspection/AnnotationDeprecatedInspection.java
+++ b/src/main/java/de/espend/idea/php/annotation/inspection/AnnotationDeprecatedInspection.java
@@ -1,7 +1,6 @@
 package de.espend.idea.php.annotation.inspection;
 
 import com.intellij.codeInspection.LocalInspectionTool;
-import com.intellij.codeInspection.LocalQuickFix;
 import com.intellij.codeInspection.ProblemHighlightType;
 import com.intellij.codeInspection.ProblemsHolder;
 import com.intellij.psi.PsiElement;
@@ -12,10 +11,8 @@ import com.jetbrains.php.lang.psi.elements.PhpClass;
 import de.espend.idea.php.annotation.util.AnnotationUtil;
 import org.jetbrains.annotations.NotNull;
 
-import java.util.Collection;
-
 public class AnnotationDeprecatedInspection extends LocalInspectionTool {
-    public static final String MESSAGE = "Annotation is deprecated";
+    public static final String MESSAGE = "[Annotations] Annotation is deprecated";
 
     @NotNull
     @Override
diff --git a/src/main/java/de/espend/idea/php/annotation/inspection/AnnotationDocBlockClassConstantNotFoundInspection.java b/src/main/java/de/espend/idea/php/annotation/inspection/AnnotationDocBlockClassConstantNotFoundInspection.java
index 0efad91..dea9062 100644
--- a/src/main/java/de/espend/idea/php/annotation/inspection/AnnotationDocBlockClassConstantNotFoundInspection.java
+++ b/src/main/java/de/espend/idea/php/annotation/inspection/AnnotationDocBlockClassConstantNotFoundInspection.java
@@ -23,7 +23,7 @@ import java.util.function.Function;
  * @author Daniel Espendiller <daniel@espendiller.net>
  */
 public class AnnotationDocBlockClassConstantNotFoundInspection extends LocalInspectionTool {
-    public static final String MESSAGE = "Class not found";
+    public static final String MESSAGE = "[Annotations] Class not found";
 
     @NotNull
     @Override
diff --git a/src/main/java/de/espend/idea/php/annotation/inspection/AnnotationDocBlockTagClassNotFoundInspection.java b/src/main/java/de/espend/idea/php/annotation/inspection/AnnotationDocBlockTagClassNotFoundInspection.java
index c73d96a..bb2e38c 100644
--- a/src/main/java/de/espend/idea/php/annotation/inspection/AnnotationDocBlockTagClassNotFoundInspection.java
+++ b/src/main/java/de/espend/idea/php/annotation/inspection/AnnotationDocBlockTagClassNotFoundInspection.java
@@ -25,7 +25,7 @@ import java.util.function.Function;
  * @author Daniel Espendiller <daniel@espendiller.net>
  */
 public class AnnotationDocBlockTagClassNotFoundInspection extends LocalInspectionTool {
-    public static final String MESSAGE = "Class not found";
+    public static final String MESSAGE = "[Annotations] Class not found";
 
     @NotNull
     @Override
diff --git a/src/main/java/de/espend/idea/php/annotation/inspection/AnnotationMissingUseInspection.java b/src/main/java/de/espend/idea/php/annotation/inspection/AnnotationMissingUseInspection.java
index 2882404..38eedd4 100644
--- a/src/main/java/de/espend/idea/php/annotation/inspection/AnnotationMissingUseInspection.java
+++ b/src/main/java/de/espend/idea/php/annotation/inspection/AnnotationMissingUseInspection.java
@@ -10,7 +10,6 @@ import com.jetbrains.php.lang.documentation.phpdoc.parser.PhpDocElementTypes;
 import com.jetbrains.php.lang.documentation.phpdoc.psi.PhpDocComment;
 import com.jetbrains.php.lang.documentation.phpdoc.psi.tags.PhpDocTag;
 import com.jetbrains.php.lang.psi.elements.PhpClass;
-import com.jetbrains.php.lang.psi.elements.PhpNamedElement;
 import de.espend.idea.php.annotation.inspection.visitor.PhpDocTagWithUsePsiElementVisitor;
 import de.espend.idea.php.annotation.util.AnnotationUtil;
 import org.apache.commons.lang.StringUtils;
@@ -29,7 +28,7 @@ import java.util.stream.Collectors;
  * @author Daniel Espendiller <daniel@espendiller.net>
  */
 public class AnnotationMissingUseInspection extends LocalInspectionTool {
-    public static final String MESSAGE = "Missing import";
+    public static final String MESSAGE = "[Annotations] Missing import";
 
     @NotNull
     @Override
diff --git a/src/main/resources/META-INF/plugin.xml b/src/main/resources/META-INF/plugin.xml
index eec82ef..1b4d3a0 100644
--- a/src/main/resources/META-INF/plugin.xml
+++ b/src/main/resources/META-INF/plugin.xml
@@ -100,42 +100,42 @@
 
       <codeInsight.lineMarkerProvider language="PHP" implementationClass="de.espend.idea.php.annotation.navigation.AnnotationUsageLineMarkerProvider"/>
 
-      <localInspection groupPath="PHP" shortName="AnnotationMissingUseInspection" displayName="Missing import"
+      <localInspection groupPath="PHP" shortName="AnnotationMissingUseInspection" displayName="[Annotations] Missing import"
                        groupName="Annotation"
                        enabledByDefault="true"
                        level="WARNING"
                        language="PHP"
                        implementationClass="de.espend.idea.php.annotation.inspection.AnnotationMissingUseInspection"/>
 
-      <localInspection groupPath="PHP" shortName="AnnotationDeprecatedInspection" displayName="Annotation is deprecated"
+      <localInspection groupPath="PHP" shortName="AnnotationDeprecatedInspection" displayName="[Annotations] Annotation is deprecated"
                        groupName="Annotation"
                        enabledByDefault="true"
                        level="WARNING"
                        language="PHP"
                        implementationClass="de.espend.idea.php.annotation.inspection.AnnotationDeprecatedInspection"/>
 
-      <localInspection groupPath="PHP" shortName="RepositoryClassInspection" displayName="Missing repository class"
+      <localInspection groupPath="PHP" shortName="RepositoryClassInspection" displayName="[Annotations] Missing repository class"
                        groupName="Annotation"
                        enabledByDefault="true"
                        level="WARNING"
                        language="PHP"
                        implementationClass="de.espend.idea.php.annotation.doctrine.inspection.RepositoryClassInspection"/>
 
-      <localInspection groupPath="PHP" shortName="DoctrineTypeDeprecatedInspection" displayName="Doctrine column type deprecated"
+      <localInspection groupPath="PHP" shortName="DoctrineTypeDeprecatedInspection" displayName="[Annotations] Doctrine column type deprecated"
                        groupName="Annotation"
                        enabledByDefault="true"
                        level="WARNING"
                        language="PHP"
                        implementationClass="de.espend.idea.php.annotation.doctrine.inspection.DoctrineTypeDeprecatedInspection"/>
 
-      <localInspection groupPath="PHP" shortName="AnnotationDocBlockTagClassNotFound" displayName="Class not found"
+      <localInspection groupPath="PHP" shortName="AnnotationDocBlockTagClassNotFound" displayName="[Annotations] Class not found"
                        groupName="Annotation"
                        enabledByDefault="true"
                        level="WARNING"
                        language="PHP"
                        implementationClass="de.espend.idea.php.annotation.inspection.AnnotationDocBlockTagClassNotFoundInspection"/>
 
-      <localInspection groupPath="PHP" shortName="AnnotationDocBlockClassConstantNotFound" displayName="Class not found"
+      <localInspection groupPath="PHP" shortName="AnnotationDocBlockClassConstantNotFound" displayName="[Annotations] Class not found"
                        groupName="Annotation"
                        enabledByDefault="true"
                        level="WARNING"
