diff --git a/src/test/java/com/googlecode/aviator/test/function/GrammarUnitTest.java b/src/test/java/com/googlecode/aviator/test/function/GrammarUnitTest.java
index a0aad19..140da70 100644
--- a/src/test/java/com/googlecode/aviator/test/function/GrammarUnitTest.java
+++ b/src/test/java/com/googlecode/aviator/test/function/GrammarUnitTest.java
@@ -20,6 +20,7 @@ import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
 import java.math.BigDecimal;
+import java.math.BigInteger;
 import java.util.ArrayList;
 import java.util.Date;
 import java.util.HashMap;
@@ -51,6 +52,18 @@ public class GrammarUnitTest {
     AviatorEvaluator.setOption(Options.ALWAYS_PARSE_FLOATING_POINT_NUMBER_INTO_DECIMAL, false);
   }
 
+
+  @Test
+  public void testIssue162() {
+    Object val = AviatorEvaluator.execute("2017122615550747128008704");
+    assertTrue(val instanceof BigInteger);
+    assertEquals(new BigInteger("2017122615550747128008704"), val);
+
+    val = AviatorEvaluator.execute("-2017122615550747128008704");
+    assertTrue(val instanceof BigInteger);
+    assertEquals(new BigInteger("-2017122615550747128008704"), val);
+  }
+
   @Test
   public void testIssue87() {
     AviatorEvaluator.setOption(Options.ALWAYS_PARSE_FLOATING_POINT_NUMBER_INTO_DECIMAL, true);
