diff --git a/src/test/java/com/projecty/projectyweb/task/TaskControllerTests.java b/src/test/java/com/projecty/projectyweb/task/TaskControllerTests.java
index 5e90f95..ebab261 100644
--- a/src/test/java/com/projecty/projectyweb/task/TaskControllerTests.java
+++ b/src/test/java/com/projecty/projectyweb/task/TaskControllerTests.java
@@ -192,9 +192,7 @@ public class TaskControllerTests {
     @Test
     @WithMockUser
     public void givenRequestOnRemoveAssignment_shouldReturnOk() throws Exception {
-        mockMvc.perform(delete("/tasks/1/assign").with(csrf())
-                .contentType(MediaType.APPLICATION_JSON)
-                .content("{\"username\":\"user1\"}"))
+        mockMvc.perform(delete("/tasks/1/assign/user1").with(csrf()))
                 .andExpect(status().isOk());
     }
 }
