diff --git a/src/test/java/com/github/pedrovgs/problem3/SumBinaryNumberTest.java b/src/test/java/com/github/pedrovgs/problem3/SumBinaryNumberTest.java
index 66861f3..9ecb134 100644
--- a/src/test/java/com/github/pedrovgs/problem3/SumBinaryNumberTest.java
+++ b/src/test/java/com/github/pedrovgs/problem3/SumBinaryNumberTest.java
@@ -156,7 +156,7 @@ public class SumBinaryNumberTest {
 
     String result = sumBinaryNumbers.sumBinaryNumbersBasedOnCount(n1, n2);
 
-    assertEquals("010", result);
+    assertEquals("10", result);
   }
 
   @Test public void zeroPlusTwoEqualsTwoBasedOnCount() {
