diff --git a/src/main/java/io/openapitools/swagger/OpenAPISorter.java b/src/main/java/io/openapitools/swagger/OpenAPISorter.java
index 98db422..716030e 100644
--- a/src/main/java/io/openapitools/swagger/OpenAPISorter.java
+++ b/src/main/java/io/openapitools/swagger/OpenAPISorter.java
@@ -61,7 +61,7 @@ public class OpenAPISorter {
             return;
         }
 
-        sortSchemas(components.getSchemas());
+        components.setSchemas(sortSchemas(components.getSchemas()));
 
         components.setResponses(createSorted(components.getResponses()));
         components.setParameters(createSorted(components.getParameters()));
