diff --git a/internal/gengapic/doc_file.go b/internal/gengapic/doc_file.go
index 136965c..6001b68 100644
--- a/internal/gengapic/doc_file.go
+++ b/internal/gengapic/doc_file.go
@@ -66,6 +66,11 @@ func (g *generator) genDocFile(year int, scopes []string, serv *descriptor.Servi
 	}
 
 	p("//")
+	p("// General documentation")
+	p("//")
+	p("// For information about setting deadlines, reusing contexts, and more")
+	p("// please visit https://pkg.go.dev/cloud.google.com/go.")
+	p("//")
 	p("// Example usage")
 	p("//")
 	p("// To get started with this package, create a client.")
@@ -103,9 +108,6 @@ func (g *generator) genDocFile(year int, scopes []string, serv *descriptor.Servi
 	p("// Individual methods on the client use the ctx given to them.")
 	p("//")
 	p("// To close the open connection, use the Close() method.")
-	p("//")
-	p("// For information about setting deadlines, reusing contexts, and more")
-	p("// please visit https://pkg.go.dev/cloud.google.com/go.")
 	p("package %s // import %q", g.opts.pkgName, g.opts.pkgPath)
 	p("")
 
