diff --git a/src/test/java/com/github/davidmoten/rx/jdbc/DatabaseTestBase.java b/src/test/java/com/github/davidmoten/rx/jdbc/DatabaseTestBase.java
index 2896ac9..17247b7 100644
--- a/src/test/java/com/github/davidmoten/rx/jdbc/DatabaseTestBase.java
+++ b/src/test/java/com/github/davidmoten/rx/jdbc/DatabaseTestBase.java
@@ -1599,7 +1599,7 @@ public abstract class DatabaseTestBase {
                 .getAs(Integer.class)
                 //
                 .toList().toBlocking().single();
-        assertEquals(Arrays.asList(2), list);
+        assertEquals(Arrays.asList(1, 2), list);
     }
 
     @Test
